--- origsrc/global-6.2.12/gtags.conf	2014-03-31 13:45:26.000000000 +0900
+++ src/global-6.2.12/gtags.conf	2014-04-09 12:20:16.256148100 +0900
@@ -51,7 +51,7 @@ builtin-parser:\
 user-custom|User custom plugin parser:\
 	:tc=common:\
 	:langmap=c\:.c.h:\
-	:gtags_parser=c\:/usr/local/lib/gtags/user-custom.la:
+	:gtags_parser=c\:/usr/lib/gtags/user-custom.la:
 #
 # Plug-in parser to use Exuberant Ctags.
 #
@@ -96,45 +96,45 @@ exuberant-ctags|plugin-example|setting t
 	:langmap=VHDL\:.vhdl.vhd:\
 	:langmap=Vim\:.vim:\
 	:langmap=YACC\:.y:\
-	:gtags_parser=Asm\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Asp\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Awk\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Basic\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=BETA\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=C\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=C++\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=C#\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Cobol\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=DosBatch\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Eiffel\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Erlang\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Flex\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Fortran\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=HTML\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Java\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=JavaScript\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Lisp\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Lua\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=MatLab\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=OCaml\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Pascal\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Perl\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=PHP\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Python\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=REXX\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Ruby\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Scheme\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Sh\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=SLang\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=SML\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=SQL\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Tcl\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Tex\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Vera\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Verilog\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=VHDL\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=Vim\:/usr/local/lib/gtags/exuberant-ctags.la:\
-	:gtags_parser=YACC\:/usr/local/lib/gtags/exuberant-ctags.la:
+	:gtags_parser=Asm\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Asp\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Awk\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Basic\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=BETA\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=C\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=C++\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=C#\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Cobol\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=DosBatch\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Eiffel\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Erlang\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Flex\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Fortran\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=HTML\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Java\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=JavaScript\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Lisp\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Lua\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=MatLab\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=OCaml\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Pascal\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Perl\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=PHP\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Python\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=REXX\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Ruby\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Scheme\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Sh\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=SLang\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=SML\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=SQL\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Tcl\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Tex\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Vera\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Verilog\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=VHDL\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=Vim\:/usr/lib/gtags/exuberant-ctags.la:\
+	:gtags_parser=YACC\:/usr/lib/gtags/exuberant-ctags.la:
 #
 # Drupal configuration.
 #
--- origsrc/global-6.2.12/htags/completion.cgi.tmpl	2014-03-31 13:45:27.000000000 +0900
+++ src/global-6.2.12/htags/completion.cgi.tmpl	2014-04-09 12:20:18.697392200 +0900
@@ -1,4 +1,4 @@
-#! /opt/local/bin/perl
+#! /usr/bin/perl
 #
 # Copyright (c) 2010 Tama Communications Corporation
 #
--- origsrc/global-6.2.12/libltdl/Makefile.am	2014-03-31 13:45:11.000000000 +0900
+++ src/global-6.2.12/libltdl/Makefile.am	2014-04-09 12:17:00.248549300 +0900
@@ -137,12 +137,6 @@ CLEANFILES		       += libltdl.la \
 CLEANFILES	       += $(ltdl_LIBOBJS) $(ltdl_LTLIBOBJS)
 
 EXTRA_DIST	       += COPYING.LIB \
-			   \
-			   \
-			   \
-			   \
-			   \
-			   \
 			  README
 
 ## --------------------------- ##
@@ -157,6 +151,7 @@ EXTRA_DIST	+= argz_.h \
 # doesn't have one that works with the given compiler.
 all-local $(lib_OBJECTS): $(ARGZ_H)
 argz.h: argz_.h
+	$(mkinstalldirs) . 
 	cp $(srcdir)/argz_.h $@-t
 	mv $@-t $@
 MOSTLYCLEANFILES += argz.h \
--- origsrc/global-6.2.12/libltdl/README	2012-11-18 13:27:16.000000000 +0900
+++ src/global-6.2.12/libltdl/README	2014-04-09 12:16:50.680592600 +0900
@@ -1,15 +1,15 @@
 This is GNU libltdl, a system independent dlopen wrapper for GNU libtool.
 
 It supports the following dlopen interfaces:
-* dlopen (POSIX)
+* dlopen (Solaris, Linux and various BSD flavors)
 * shl_load (HP-UX)
 * LoadLibrary (Win16 and Win32)
 * load_add_on (BeOS)
 * GNU DLD (emulates dynamic linking for static libraries)
 * dyld (darwin/Mac OS X)
 * libtool's dlpreopen
---
-   Copyright (C) 1999, 2003, 2011 Free Software Foundation, Inc.
+-- 
+   Copyright (C) 1999, 2003 Free Software Foundation, Inc.
    Written by Thomas Tanner, 1999
 
    This file is part of GNU Libtool.
--- origsrc/global-6.2.12/libltdl/ltdl.c	2012-11-18 13:27:16.000000000 +0900
+++ src/global-6.2.12/libltdl/ltdl.c	2014-04-09 12:16:59.325457000 +0900
@@ -1,7 +1,7 @@
 /* ltdl.c -- system independent dlopen wrapper
 
    Copyright (C) 1998, 1999, 2000, 2004, 2005, 2006,
-		 2007, 2008, 2011 Free Software Foundation, Inc.
+		 2007, 2008 Free Software Foundation, Inc.
    Written by Thomas Tanner, 1998
 
    NOTE: The canonical source of this file is maintained with the
@@ -80,11 +80,6 @@ static  const char	libprefix[]		= LT_LIB
 #if defined(LT_MODULE_EXT)
 static	const char	shlib_ext[]		= LT_MODULE_EXT;
 #endif
-/* If the loadable module suffix is not the same as the linkable
- * shared library suffix, this will be defined. */
-#if defined(LT_SHARED_EXT)
-static	const char	shared_ext[]		= LT_SHARED_EXT;
-#endif
 #if defined(LT_DLSEARCH_PATH)
 static	const char	sys_dlsearch_path[]	= LT_DLSEARCH_PATH;
 #endif
@@ -1542,9 +1537,6 @@ has_library_ext (const char *filename)
 #if defined(LT_MODULE_EXT)
 	     || (streq (ext, shlib_ext))
 #endif
-#if defined(LT_SHARED_EXT)
-	     || (streq (ext, shared_ext))
-#endif
     ))
     {
       return 1;
@@ -1687,17 +1679,6 @@ lt_dlopenadvise (const char *filename, l
 
       /* As before, if the file was found but loading failed, return now
 	 with the current error message.  */
-      if (handle || ((errors > 0) && !file_not_found ()))
-	return handle;
-#endif
-
-#if defined(LT_SHARED_EXT)
-      /* Try appending SHARED_EXT.   */
-      LT__SETERRORSTR (saved_error);
-      errors = try_dlopen (&handle, filename, shared_ext, advise);
-
-      /* As before, if the file was found but loading failed, return now
-	 with the current error message.  */
       if (handle || ((errors > 0) && !file_not_found ()))
 	return handle;
 #endif
--- origsrc/global-6.2.12/plugin-factory/Makefile.am	2014-03-31 13:45:06.000000000 +0900
+++ src/global-6.2.12/plugin-factory/Makefile.am	2014-04-09 12:10:05.507902200 +0900
@@ -14,11 +14,11 @@ plugindir = ${libdir}/gtags
 plugin_LTLIBRARIES = exuberant-ctags.la user-custom.la
 
 exuberant_ctags_la_SOURCES = exuberant-ctags.c
-exuberant_ctags_la_LDFLAGS = -module -avoid-version
+exuberant_ctags_la_LDFLAGS = -module -avoid-version -no-undefined
 
 # skeleton
 user_custom_la_SOURCES = user-custom.c
-user_custom_la_LDFLAGS = -module -avoid-version
+user_custom_la_LDFLAGS = -module -avoid-version -no-undefined
 
 gtagsdir = ${datadir}/gtags
 gtags_DATA = PLUGIN_HOWTO
