--- origsrc/less-382/CYGWIN-PATCHES/README	1970-01-01 09:00:00.000000000 +0900
+++ src/less-382/CYGWIN-PATCHES/README	2018-05-07 14:57:30.812500100 +0900
@@ -0,0 +1,77 @@
+less-382-iso262-ext03
+------------------------------------------------------------------------
+Enhanced less for reading texts using ISO 2022 code extention techniques 
+and using multiple Japanese encoding schemes.
+
+Runtime requirements:
+  cygwin
+  libncursesw10
+
+Build requirements:
+  cygport-usrlocal-0.8-1
+  cygwin
+  libncursesw-devel
+  libncursesw10
+
+Canonical homepage:
+  Original:    http://www.greenwoodsoftware.com/less/
+  Jam less:    http://www25.big.jp/~jam/less/ (gone) -> http://ftp.jaist.ac.jp/pub/FreeBSD/ports/disitfiles
+  UTF-8/CP932: http://bogytech.blogspot.jp/search/label/jless
+
+Canonical download:
+  http://ftp.jaist.ac.jp/pub/FreeBSD/ports/distfiles/less-382.tar.gz
+  http://ftp.jaist.ac.jp/pub/FreeBSD/ports/distfiles/regex_cs-lwp9k.shar.gz
+  http://ftp.jaist.ac.jp/pub/FreeBSD/ports/distfiles/less-382-iso258.patch.gz
+  http://ftp.jaist.ac.jp/pub/FreeBSD/ports/distfiles/less-382-iso259-260.patch.gz
+  http://ftp.jaist.ac.jp/pub/FreeBSD/ports/distfiles/less-382-iso260-261.patch.gz
+  http://ftp.jaist.ac.jp/pub/FreeBSD/ports/distfiles/less-382-iso261-262.patch.gz
+  http://ftp.jaist.ac.jp/pub/FreeBSD/ports/distfiles/less-382-iso262.ext03.patch.gz
+
+License:
+  Less License (BSD style license)
+
+Language:
+  C
+
+------------------------------------------------------------------------
+
+Build instructions:
+  unpack less-382-iso262-ext03-cs_regex-3-src.tar.xz
+    if you use setup to install this src package, it will be
+	 unpacked under /usr/src automatically
+  If you want to set prefix as /usr/local, set environment variable USRLOCAL.
+  cd /usr/src
+  cygport jless.cygport all
+
+This will create:
+  less-382-iso262-ext03-cs_regex-3.tar.xz
+  less-382-iso262-ext03-cs_regex-3-src.tar.xz
+
+Or use 'cygport jless.cygport prep' to get a patched source directory
+
+To find out the files included in the binary distribution, you can
+use "cygcheck -l jless".
+
+------------------------------------------------------------------------
+
+Port Notes:
+
+----------  less-382-iso262-ext03-cs_regex-3 -- 2018/05/07 -------------
+- Adjust and rebuild for the latest Cygwin environment.
+
+----------  less-382-iso262-ext03-cs_regex-2 -- 2014/05/05 -------------
+- Build for 64bit also.
+
+----------  less-382-iso262-ext03-cs_regex-1 -- 2014/04/29 -------------
+- Use ext03 patch instead of CP932 patch.
+
+----------  less-382-iso262-imd1-2 -- 2008/01/01 -----------------------
+Initial release.
+- Use cygport framework.
+
+For more information about this package, see the upstream documentation in
+/usr/share/doc/less.
+
+This private port maintained by: Yak! <yak_ex@mx.scn.tv>
+Please address all questions to the above address, NOT to the Cygwin 
+mailing list at <cygwin@cygwin.com>.
--- origsrc/less-382/CYGWIN-PATCHES/setup.hint	1970-01-01 09:00:00.000000000 +0900
+++ src/less-382/CYGWIN-PATCHES/setup.hint	2018-05-07 14:56:16.729092500 +0900
@@ -0,0 +1,7 @@
+@ jless
+sdesc: "Jam less"
+ldesc: "Enhanced less for reading texts using ISO 2022 code extention techniques and using multiple Japanese encoding schemes."
+category: Utils Text
+requires: cygwin libncursesw10
+curr: 382-iso262-ext03-cs_regex-3
+prev: 382-iso262-ext03-cs_regex-2
