--- origsrc/libiconv-1.14/Makefile.devel	2011-07-03 23:05:11.000000000 +0900
+++ src/libiconv-1.14/Makefile.devel	2014-04-11 12:46:37.030762800 +0900
@@ -4,8 +4,8 @@
 
 SHELL = /bin/sh
 MAKE = make
-AUTOCONF = autoconf-2.68
-AUTOHEADER = autoheader-2.68
+AUTOCONF = autoconf-2.69
+AUTOHEADER = autoheader-2.69
 AUTOMAKE = automake-1.11
 ACLOCAL = aclocal-1.11
 GPERF = gperf
@@ -190,8 +190,8 @@ lib/flags.h : lib/encodings.def lib/enco
 
 
 lib/translit.h : lib/translit.def lib/gentranslit.c
-	$(CC) $(CFLAGS) lib/gentranslit.c -o gentranslit
-	ulimit -s unlimited; ./gentranslit < lib/translit.def > lib/translit.h
+	$(CC) $(CFLAGS) -Wl,--stack,16777216  lib/gentranslit.c -o gentranslit
+	./gentranslit < lib/translit.def > lib/translit.h
 	$(RM) gentranslit
 
 
--- origsrc/libiconv-1.14/configure.ac	2011-08-08 02:48:03.000000000 +0900
+++ src/libiconv-1.14/configure.ac	2014-05-05 17:00:25.321419200 +0900
@@ -20,6 +20,7 @@ AC_PREREQ([2.60])
 AC_INIT
 AC_CONFIG_SRCDIR([lib/iconv.c])
 AC_CONFIG_AUX_DIR([build-aux])
+AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([libiconv], [1.14])
 AC_CONFIG_HEADERS([config.h lib/config.h])
 AC_PROG_MAKE_SET
--- origsrc/libiconv-1.14/libcharset/Makefile.devel	2010-09-24 08:28:15.000000000 +0900
+++ src/libiconv-1.14/libcharset/Makefile.devel	2014-04-11 12:46:37.050764800 +0900
@@ -3,8 +3,8 @@
 
 SHELL = /bin/sh
 MAKE = make
-AUTOCONF = autoconf-2.68
-AUTOHEADER = autoheader-2.68
+AUTOCONF = autoconf-2.69
+AUTOHEADER = autoheader-2.69
 ACLOCAL = aclocal-1.11
 CP = cp
 RM = rm -f
--- origsrc/libiconv-1.14/libcharset/configure.ac	2009-10-19 00:26:59.000000000 +0900
+++ src/libiconv-1.14/libcharset/configure.ac	2014-05-05 17:00:52.820996900 +0900
@@ -23,6 +23,7 @@ VERSION=1.4
 
 AC_INIT([lib/localcharset.c])
 AC_CONFIG_AUX_DIR([build-aux])
+AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADER([config.h])
 AC_PROG_MAKE_SET
 AC_SUBST([PACKAGE])
--- origsrc/libiconv-1.14/preload/Makefile.devel	2010-09-24 08:29:30.000000000 +0900
+++ src/libiconv-1.14/preload/Makefile.devel	2014-04-11 12:46:37.091768900 +0900
@@ -3,7 +3,7 @@
 
 SHELL = /bin/sh
 MAKE = make
-AUTOCONF = autoconf-2.68
+AUTOCONF = autoconf-2.69
 ACLOCAL = aclocal-1.11
 CP = cp
 RM = rm -f
