diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/cmd/lefty/common.h src/graphviz-2.38.0/cmd/lefty/common.h
--- origsrc/graphviz-2.38.0/cmd/lefty/common.h	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/cmd/lefty/common.h	2014-04-17 19:16:35.756824300 +0900
@@ -29,7 +29,7 @@
  */
 #define FEATURE_MINTSIZE 1 
 #define FEATURE_DOT 1
-#if defined(WIN32) || defined(MSWIN32)
+#if defined(WIN32) || defined(MSWIN32) || defined(CYGWIN_WIN32)
 #ifndef FEATURE_WIN32
 #define FEATURE_WIN32 1
 #endif
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/cmd/lefty/lefty.c src/graphviz-2.38.0/cmd/lefty/lefty.c
--- origsrc/graphviz-2.38.0/cmd/lefty/lefty.c	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/cmd/lefty/lefty.c	2014-04-17 19:16:35.785827200 +0900
@@ -207,6 +207,61 @@
 
 #else
 
+#ifdef __CYGWIN__
+static int parse_arg(char *arg, char **argv)
+{
+    enum { NORMAL, DELIMITED, QUOTED, ESCAPED, ESCAPED_QUOTED } state;
+    int num = 0;
+    state = DELIMITED;
+    while(*arg != 0) { 
+        switch(*arg) {
+        case '"':  
+            switch(state) {
+            case NORMAL: state = QUOTED;                                                         break;
+            case DELIMITED:      state = QUOTED;         if(argv) argv[num++] = arg; else ++num; break;
+            case QUOTED:         state = NORMAL;                                                 break;
+            case ESCAPED:        state = NORMAL;                                                 break;
+            case ESCAPED_QUOTED: state = QUOTED;                                                 break;
+            }
+            break;
+        case '\\':
+            switch(state) {
+            case NORMAL:         state = ESCAPED;                                                break;
+            case DELIMITED:      state = ESCAPED;        if(argv) argv[num++] = arg; else ++num; break;
+            case QUOTED:         state = ESCAPED_QUOTED;                                         break;
+            case ESCAPED:        state = NORMAL;                                                 break;
+            case ESCAPED_QUOTED: state = QUOTED;                                                 break;
+            }
+            break;
+        case ' ':
+        case '\t':
+        case '\r':
+        case '\n':
+            switch(state) {
+            case NORMAL:         state = DELIMITED;      if(argv) *arg = 0;                      break;
+            case DELIMITED:      state = DELIMITED;                                              break;
+            case QUOTED:         state = QUOTED;                                                 break;
+            case ESCAPED:        state = NORMAL;                                                 break;
+            case ESCAPED_QUOTED: state = QUOTED;                                                 break;
+            }
+            break;
+        default:
+            switch(state) {
+            case NORMAL:                                                                         break;
+            case DELIMITED:      state = NORMAL;         if(argv) argv[num++] = arg; else ++num; break;
+            case QUOTED:         state = QUOTED;                                                 break;
+            case ESCAPED:        state = NORMAL;                                                 break;
+            case ESCAPED_QUOTED: state = QUOTED;                                                 break;
+            }
+            break;
+        }
+        ++arg;
+    }
+    if(argv) argv[num] = 0;
+    return num;
+}
+#endif
+
 HINSTANCE hinstance, hprevinstance;
 
 int APIENTRY WinMain (
@@ -214,6 +269,10 @@
 ) {
     Tobj co;
     Psrc_t src;
+#ifdef __CYGWIN__
+    char **__argv, *buf;
+    int __argc;
+#endif
 
     hinstance = hInstance;
     hprevinstance = hPrevInstance;
@@ -231,6 +290,12 @@
     Eshowbody = 1;
     Eshowcalls = 1;
 
+#ifdef __CYGWIN__
+    buf = strdup(GetCommandLine());
+    __argc = parse_arg(buf, NULL);
+    __argv = malloc(sizeof(char*) * (__argc + 1));
+    parse_arg(buf, __argv);
+#endif
     processstr (leftyoptions);
     __argv++, __argc--;
     processargs (__argc, __argv);
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/lib/common/globals.c src/graphviz-2.38.0/lib/common/globals.c
--- origsrc/graphviz-2.38.0/lib/common/globals.c	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/lib/common/globals.c	2014-04-17 19:16:35.859834600 +0900
@@ -15,8 +15,9 @@
 #include "config.h"
 #endif
 
-#define EXTERN
 #include "types.h"
+#define EXTERN __declspec(dllexport)
+#define extern __declspec(dllexport)
 #include "globals.h"
 #include "fdp.h"
 
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/lib/common/timing.c src/graphviz-2.38.0/lib/common/timing.c
--- origsrc/graphviz-2.38.0/lib/common/timing.c	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/lib/common/timing.c	2014-04-17 19:16:35.890837700 +0900
@@ -42,11 +42,13 @@
 
 static mytime_t T;
 
+__declspec(dllexport)
 void start_timer(void)
 {
     GET_TIME(T);
 }
 
+__declspec(dllexport)
 double elapsed_sec(void)
 {
     mytime_t S;
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/lib/gvc/gvcext.h src/graphviz-2.38.0/lib/gvc/gvcext.h
--- origsrc/graphviz-2.38.0/lib/gvc/gvcext.h	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/lib/gvc/gvcext.h	2014-04-17 19:20:06.863932900 +0900
@@ -76,7 +76,7 @@
 
 
 #ifndef WIN32_DLL
-#if defined(GVDLL)
+#if defined(GVDLL) && !defined(__CYGWIN__)
 	__declspec(dllexport) lt_symlist_t lt_preloaded_symbols[];
 #else
 #if !defined(LTDL_H)
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/lib/gvpr/compile.c src/graphviz-2.38.0/lib/gvpr/compile.c
--- origsrc/graphviz-2.38.0/lib/gvpr/compile.c	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/lib/gvpr/compile.c	2014-04-17 19:16:35.956844300 +0900
@@ -74,7 +74,7 @@
 
 static Agiodisc_t gprIoDisc = { iofread, ioputstr, ioflush };
 
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
 static Agdisc_t gprDisc = { 0, 0, &gprIoDisc };
 #else
 static Agdisc_t gprDisc = { &AgMemDisc, &AgIdDisc, &gprIoDisc };
@@ -2555,7 +2555,7 @@
 {
     Agraph_t *g;
 
-#ifdef WIN32
+#if defined(WIN32) || defined(__CYGWIN__)
     gprDisc.mem = &AgMemDisc;
     gprDisc.id = &AgIdDisc;
 #endif
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/lib/gvpr/gvpr.c src/graphviz-2.38.0/lib/gvpr/gvpr.c
--- origsrc/graphviz-2.38.0/lib/gvpr/gvpr.c	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/lib/gvpr/gvpr.c	2014-04-17 19:16:35.984847100 +0900
@@ -938,6 +938,9 @@
  *  - do automatic cast for array indices if type is known
  *  - array initialization
  */
+#ifdef __CYGWIN__
+__declspec(dllexport)
+#endif
 int gvpr (int argc, char *argv[], gvpropts * uopts)
 {
     Sfdisc_t errdisc;
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/lib/vmalloc/features/vmalloc src/graphviz-2.38.0/lib/vmalloc/features/vmalloc
--- origsrc/graphviz-2.38.0/lib/vmalloc/features/vmalloc	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/lib/vmalloc/features/vmalloc	2014-04-17 19:27:19.656207800 +0900
@@ -21,6 +21,7 @@
 
 std	malloc note{ stuck with standard malloc }end noexecute{
 	_BEGIN_EXTERNS_
+	#if 0
 	extern void _exit _ARG_((int));
 	extern char* strdup _ARG_((const char*));
 	#if _STD_
@@ -28,8 +29,9 @@
 	#else
 	char* malloc(n) unsigned n; { _exit(0); return 0; }
 	#endif
+	#endif
 	_END_EXTERNS_
-	main() { strdup("yo"); _exit(1); }
+	main() { /* strdup("yo"); */ _exit(1); } /* FORCE TO STUCK WITH STD */
 }end
 
 lib alloca note{ alloca exists }end compile{
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/plugin/core/Makefile.am src/graphviz-2.38.0/plugin/core/Makefile.am
--- origsrc/graphviz-2.38.0/plugin/core/Makefile.am	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/plugin/core/Makefile.am	2014-04-17 20:25:53.381529700 +0900
@@ -17,7 +17,13 @@
 pkglib_LTLIBRARIES = libgvplugin_core.la
 endif
 
+if WITH_WIN32
 libgvplugin_core_C_la_SOURCES = \
+	$(top_builddir)/lib/common/libcommon_C.la
+else
+libgvplugin_core_C_la_SOURCES = 
+endif
+libgvplugin_core_C_la_SOURCES += \
 	gvplugin_core.c \
 	gvrender_core_dot.c \
 	gvrender_core_fig.c \
@@ -44,7 +50,6 @@
 
 if WITH_WIN32
 libgvplugin_core_la_LDFLAGS += -no-undefined
-libgvplugin_core_la_LIBADD += $(top_builddir)/lib/cgraph/libcgraph.la
 endif
 
 if WITH_DARWIN9
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/plugin/dot_layout/gvplugin_dot_layout.c src/graphviz-2.38.0/plugin/dot_layout/gvplugin_dot_layout.c
--- origsrc/graphviz-2.38.0/plugin/dot_layout/gvplugin_dot_layout.c	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/plugin/dot_layout/gvplugin_dot_layout.c	2014-04-17 19:16:36.098858500 +0900
@@ -12,6 +12,7 @@
  *************************************************************************/
 
 #include "gvplugin.h"
+#include "config.h"
 
 extern gvplugin_installed_t gvlayout_dot_layout[];
 
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/plugin/gd/Makefile.am src/graphviz-2.38.0/plugin/gd/Makefile.am
--- origsrc/graphviz-2.38.0/plugin/gd/Makefile.am	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/plugin/gd/Makefile.am	2014-04-17 20:28:23.546565700 +0900
@@ -40,7 +40,7 @@
 
 if WITH_WIN32
 libgvplugin_gd_la_LDFLAGS += -no-undefined
-libgvplugin_gd_la_LIBADD += $(top_builddir)/lib/cgraph/libcgraph.la $(top_builddir)/lib/pathplan/libpathplan.la
+libgvplugin_gd_la_LIBADD += $(top_builddir)/lib/pathplan/libpathplan.la $(top_builddir)/plugin/core/libgvplugin_core.la
 endif
 
 if WITH_DARWIN9
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/plugin/neato_layout/gvplugin_neato_layout.c src/graphviz-2.38.0/plugin/neato_layout/gvplugin_neato_layout.c
--- origsrc/graphviz-2.38.0/plugin/neato_layout/gvplugin_neato_layout.c	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/plugin/neato_layout/gvplugin_neato_layout.c	2014-04-17 19:16:36.121860800 +0900
@@ -12,6 +12,7 @@
  *************************************************************************/
 
 #include "gvplugin.h"
+#include "config.h"
 
 extern gvplugin_installed_t gvlayout_neato_types[];
 
diff -urN -x Makefile.in -x Makefile -x autom4te.cache -x libltdl -x m4 -x 'config*' -x '*.temp' -x Makefile.orig -x '*.xml' -x aclocal.m4 origsrc/graphviz-2.38.0/tclpkg/gv/Makefile.am src/graphviz-2.38.0/tclpkg/gv/Makefile.am
--- origsrc/graphviz-2.38.0/tclpkg/gv/Makefile.am	2014-04-14 05:40:25.000000000 +0900
+++ src/graphviz-2.38.0/tclpkg/gv/Makefile.am	2014-04-17 19:22:23.876632800 +0900
@@ -26,7 +26,7 @@
 nodist_libgv_sharp_la_SOURCES = gv_sharp.cpp $(SHARP_data)
 libgv_sharp_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_sharp_la_LIBADD = $(BASELIBS) $(SHARP_LIBS) 
-libgv_sharp_la_LDFLAGS = -module -avoid-version
+libgv_sharp_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_sharp_la_CPPFLAGS = $(BASECPPFLAGS) $(SHARP_INCLUDES)
 $(SHARP_data): gv_sharp.cpp
 gv_sharp.cpp: gv.i
@@ -59,7 +59,7 @@
 nodist_libgv_guile_la_SOURCES = gv_guile.cpp $(GUILE_data)
 libgv_guile_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_guile_la_LIBADD = $(BASELIBS) $(GUILE_LIBS)
-libgv_guile_la_LDFLAGS = -module -avoid-version
+libgv_guile_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_guile_la_CPPFLAGS = $(BASECPPFLAGS) $(GUILE_INCLUDES)
 # $(GUILE_data): gv_guile.cpp
 gv_guile.cpp: gv.i
@@ -72,7 +72,7 @@
 nodist_libgv_io_la_SOURCES = gv_io.cpp $(IO_data)
 libgv_io_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_io_la_LIBADD = $(BASELIBS) $(IO_LIBS)
-libgv_io_la_LDFLAGS = -module -avoid-version
+libgv_io_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_io_la_CPPFLAGS = $(BASECPPFLAGS) $(IO_INCLUDES)
 # $(IO_data): gv_io.cpp
 gv_io.cpp: gv.i
@@ -85,7 +85,7 @@
 nodist_libgv_java_la_SOURCES = gv_java.cpp
 libgv_java_la_SOURCES = $(BASESOURCES)  gv_java_init.c
 libgv_java_la_LIBADD = $(BASELIBS) $(JAVA_LIBS)
-libgv_java_la_LDFLAGS = -module -avoid-version
+libgv_java_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_java_la_CPPFLAGS = -DSWIGJAVA $(BASECPPFLAGS) $(JAVA_INCLUDES)
 $(JAVA_data): gv_java.cpp
 gv_java.cpp: gv.i
@@ -96,7 +96,7 @@
 nodist_libgv_lua_la_SOURCES = gv_lua.cpp $(LUA_data)
 libgv_lua_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_lua_la_LIBADD = $(BASELIBS) $(LUA_LIBS)
-libgv_lua_la_LDFLAGS = -module -avoid-version
+libgv_lua_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_lua_la_CPPFLAGS = $(BASECPPFLAGS) $(LUA_INCLUDES)
 # $(LUA_data): gv_lua.cpp
 gv_lua.cpp: gv.i
@@ -114,7 +114,7 @@
 nodist_libgv_ocaml_la_SOURCES = gv_ocaml.cpp $(OCAML_data)
 libgv_ocaml_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_ocaml_la_LIBADD = $(BASELIBS) $(OCAML_LIBS)
-libgv_ocaml_la_LDFLAGS = -module -avoid-version
+libgv_ocaml_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_ocaml_la_CPPFLAGS = $(BASECPPFLAGS) $(OCAML_INCLUDES)
 gv.ml gv.mli: gv_ocaml.cpp
 gv_ocaml.cpp: gv.i
@@ -152,7 +152,7 @@
 nodist_libgv_perl_la_SOURCES = gv_perl.cpp $(PERL_data)
 libgv_perl_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_perl_la_LIBADD = $(BASELIBS) $(PERL_LIBS)
-libgv_perl_la_LDFLAGS = -module -avoid-version $(PERL_LD)
+libgv_perl_la_LDFLAGS = -module -avoid-version -no-undefined $(PERL_LD)
 libgv_perl_la_CPPFLAGS = $(BASECPPFLAGS) $(PERL_CC)
 $(PERL_data): gv_perl.cpp
 gv_perl.cpp: gv.i
@@ -163,7 +163,7 @@
 nodist_libgv_php_la_SOURCES = gv_php.cpp php_gv.h
 libgv_php_la_SOURCES = $(BASESOURCES) gv_php_init.c
 libgv_php_la_LIBADD = $(BASELIBS) $(PHP_LIBS)
-libgv_php_la_LDFLAGS = -module -avoid-version
+libgv_php_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_php_la_CPPFLAGS = $(BASECPPFLAGS) $(PHP_INCLUDES)
 gv.php: gv_php.cpp
 php_gv.h: gv_php.cpp
@@ -175,7 +175,7 @@
 nodist_libgv_python_la_SOURCES = gv_python.cpp $(PYTHON_data)
 libgv_python_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_python_la_LIBADD = $(BASELIBS) $(PYTHON_LIBS)
-libgv_python_la_LDFLAGS = -module -avoid-version
+libgv_python_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_python_la_CPPFLAGS = $(BASECPPFLAGS) $(PYTHON_INCLUDES)
 $(PYTHON_data): gv_python.cpp
 gv_python.cpp: gv.i
@@ -186,7 +186,7 @@
 nodist_libgv_python23_la_SOURCES = gv_python23.cpp $(PYTHON23_data)
 libgv_python23_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_python23_la_LIBADD = $(BASELIBS) $(PYTHON23_LIBS)
-libgv_python23_la_LDFLAGS = -module -avoid-version
+libgv_python23_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_python23_la_CPPFLAGS = $(BASECPPFLAGS) $(PYTHON23_INCLUDES)
 # $(PYTHON23_data): gv_python23.cpp
 gv_python23.cpp: gv.i
@@ -197,7 +197,7 @@
 nodist_libgv_python24_la_SOURCES = gv_python24.cpp $(PYTHON24_data)
 libgv_python24_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_python24_la_LIBADD = $(BASELIBS) $(PYTHON24_LIBS)
-libgv_python24_la_LDFLAGS = -module -avoid-version
+libgv_python24_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_python24_la_CPPFLAGS = $(BASECPPFLAGS) $(PYTHON24_INCLUDES)
 # $(PYTHON24_data): gv_python24.cpp
 gv_python24.cpp: gv.i
@@ -208,7 +208,7 @@
 nodist_libgv_python25_la_SOURCES = gv_python25.cpp $(PYTHON25_data)
 libgv_python25_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_python25_la_LIBADD = $(BASELIBS) $(PYTHON25_LIBS)
-libgv_python25_la_LDFLAGS = -module -avoid-version
+libgv_python25_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_python25_la_CPPFLAGS = $(BASECPPFLAGS) $(PYTHON25_INCLUDES)
 # $(PYTHON25_data): gv_python25.cpp
 gv_python25.cpp: gv.i
@@ -219,7 +219,7 @@
 nodist_libgv_python26_la_SOURCES = gv_python26.cpp $(PYTHON26_data)
 libgv_python26_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_python26_la_LIBADD = $(BASELIBS) $(PYTHON26_LIBS)
-libgv_python26_la_LDFLAGS = -module -avoid-version
+libgv_python26_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_python26_la_CPPFLAGS = $(BASECPPFLAGS) $(PYTHON26_INCLUDES)
 # $(PYTHON26_data): gv_python26.cpp
 gv_python26.cpp: gv.i
@@ -241,7 +241,7 @@
 nodist_libgv_R_la_SOURCES = gv_R.cpp $(R_data)
 libgv_R_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_R_la_LIBADD = $(BASELIBS) $(R_LIBS)
-libgv_R_la_LDFLAGS = -module -avoid-version
+libgv_R_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_R_la_CPPFLAGS = $(BASECPPFLAGS) $(R_CFLAGS)
 # $(R_data): gv_R.cpp
 gv_R.cpp: gv.i
@@ -252,7 +252,7 @@
 nodist_libgv_ruby_la_SOURCES = gv_ruby.cpp $(RUBY_data)
 libgv_ruby_la_SOURCES = $(BASESOURCES) gv_dummy_init.c
 libgv_ruby_la_LIBADD = $(BASELIBS) $(RUBY_LIBS)
-libgv_ruby_la_LDFLAGS = -module -avoid-version
+libgv_ruby_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_ruby_la_CPPFLAGS = $(BASECPPFLAGS) $(RUBY_CFLAGS)
 # $(RUBY_data): gv_ruby.cpp
 gv_ruby.cpp: gv.i
@@ -263,7 +263,7 @@
 nodist_libgv_tcl_la_SOURCES = gv_tcl.cpp $(TCL_data)
 libgv_tcl_la_SOURCES = $(BASESOURCES) gv_tcl_init.c
 libgv_tcl_la_LIBADD = $(top_builddir)/tclpkg/tclstubs/libtclstubs_C.la $(BASELIBS) $(TCL_LIBS)
-libgv_tcl_la_LDFLAGS = -module -avoid-version
+libgv_tcl_la_LDFLAGS = -module -avoid-version -no-undefined
 libgv_tcl_la_CPPFLAGS = $(BASECPPFLAGS) $(TCL_INCLUDES)
 # $(TCL_data): gv_tcl.cpp
 gv_tcl.cpp: gv.i
