--- origsrc/gts-0.7.6/src/Makefile.am	2005-01-19 13:48:22.000000000 +0900
+++ src/gts-0.7.6/src/Makefile.am	2010-03-10 07:46:00.433170600 +0900
@@ -13,7 +13,7 @@ gts-config: gts-config.in
 lib_LTLIBRARIES = libgts.la
 
 libgts_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)\
-	-release $(LT_RELEASE) -export-dynamic
+	-export-dynamic -no-undefined
 
 libgts_la_DEPENDENCIES = \
 	predicates_init.h
@@ -66,7 +66,7 @@ predicates.o: predicates.c predicates_in
 	$(COMPILE) -c $(srcdir)/predicates.c
 
 predicates_init: predicates_init.c rounding.h
-	$(COMPILE) $(srcdir)/predicates_init.c -o $(srcdir)/predicates_init
+	$(COMPILE) $(srcdir)/predicates_init.c -o ./predicates_init
 
 predicates_init.h: predicates_init
 	./predicates_init >  $(srcdir)/predicates_init.h
