--- origsrc/nkf-2.1.3/Makefile	2013-11-22 14:17:36.000000000 +0900
+++ src/nkf-2.1.3/Makefile	2014-04-10 12:30:18.051939100 +0900
@@ -8,6 +8,7 @@ RM = rm -rf
 VERSION = 2.1.3
 MKDIR = mkdir
 prefix = /usr/local
+mandir = /usr/local/man
 PYTHON2 = python
 PYTHON3 = python
 
@@ -44,14 +45,13 @@ python3:
 	$(PYTHON3) setup.py build )
 
 install:
-	-$(MKDIR) $(prefix)/bin
-	-$(MKDIR) $(prefix)/man
-	-$(MKDIR) $(prefix)/man/man1
-	-$(MKDIR) $(prefix)/man/ja
-	-$(MKDIR) $(prefix)/man/ja/man1
-	cp -f nkf $(prefix)/bin/
-	cp -f nkf.1 $(prefix)/man/man1/
-	cp -f nkf.1j $(prefix)/man/ja/man1/nkf.1
+	-$(MKDIR) $(mandir)
+	-$(MKDIR) $(mandir)/man1
+	-$(MKDIR) $(mandir)/ja
+	-$(MKDIR) $(mandir)/ja/man1
+	cp nkf.exe $(prefix)/bin/
+	cp nkf.1 $(mandir)/man1/
+	cp nkf.1j $(mandir)/ja/man1/nkf.1
 
 shar:
 	-mkdir nkf-$(VERSION)
