--- origsrc/nkf-2.1.3/CYGWIN-PATCHES/README	1970-01-01 09:00:00.000000000 +0900
+++ src/nkf-2.1.3/CYGWIN-PATCHES/README	2014-04-10 12:36:39.274057500 +0900
@@ -0,0 +1,69 @@
+nkf
+------------------------------------------------------------------------
+Nkf is a yet another kanji code converter among networks, 
+hosts and terminals. It converts input kanji code to designated 
+kanji code such as 7-bit JIS, MS-kanji (shifted-JIS), utf-8 or EUC.
+One of the most unique facility of nkf is the guess of the input kanji code.
+
+Runtime requirements:
+  cygwin-1.7.1-1
+  perl-5.10.1-2
+
+Build requirements:
+  cygport-usrlocal-0.4-1
+  cygwin-1.7.1-1
+  perl-5.10.1-2
+
+Canonical homepage:
+  http://sourceforge.jp/projects/nkf/
+
+Canonical download:
+  http://osdn.dl.sourceforge.jp/nkf/59912/nkf-2.1.3.tar.gz
+
+License:
+  zlib/libpng License
+
+Language:
+  C
+
+------------------------------------------------------------------------
+
+Build instructions:
+  unpack nkf-2.1.3-1.tar.bz2
+    if you use setup to install this src package, it will be
+	 unpacked under /usr/src automatically
+  If you want to set prefix as /usr/local, set environment variable USRLOCAL.
+  cd /usr/src
+  cygport nkf-2.1.3-1.cygport all
+
+This will create:
+  nkf-2.1.3-1.tar.bz2
+  nkf-perl-2.1.3-1.tar.bz2
+  nkf-2.1.3-1.tar.bz2
+
+Or use 'cygport nkf-2.1.3-1.cygport prep' to get a patched source directory
+
+To find out the files included in the binary distribution, you can
+use "cygcheck -l nkf".
+
+------------------------------------------------------------------------
+
+Port Notes:
+
+----------  nkf-2.1.3-1 -- 2014/04/10 ----------------------------
+Upgrade to upstream version 2.1.3.
+
+----------  nkf-2.1.0-1 -- 2009/12/30 ----------------------------
+Upgrade to upstream version 2.1.0.
+- Rebuild for Cygwin 1.7.
+
+----------  nkf-2.0.9-1 -- 2009/08/14 ----------------------------
+Immigrate to cygport framework.
+- Import changes until 03 Aug 2009 (tree 548a10a7c9a6cf51f1e2d714f57078e29727ea3a).
+
+For more information about this package, see the upstream documentation in
+/usr/share/doc/nkf-2.1.0-1.
+
+This private port maintained by: Yak! <yak_ex@mx.scn.tv>
+Please address all questions to the above address,
+NOT to the Cygwin mailing list.
--- origsrc/nkf-2.1.3/CYGWIN-PATCHES/nkf.hint	1970-01-01 09:00:00.000000000 +0900
+++ src/nkf-2.1.3/CYGWIN-PATCHES/nkf.hint	2014-04-10 12:37:02.578387700 +0900
@@ -0,0 +1,8 @@
+@ nkf
+sdesc: "Network Kanji Filter"
+ldesc: "Nkf is a yet another kanji code converter among networks, 
+hosts and terminals. It converts input kanji code to designated 
+kanji code such as 7-bit JIS, MS-kanji (shifted-JIS), utf-8 or EUC.
+One of the most unique facility of nkf is the guess of the input kanji code."
+category: Utils
+requires: cygwin
--- origsrc/nkf-2.1.3/CYGWIN-PATCHES/perl.hint	1970-01-01 09:00:00.000000000 +0900
+++ src/nkf-2.1.3/CYGWIN-PATCHES/perl.hint	2014-04-10 12:37:09.233053100 +0900
@@ -0,0 +1,8 @@
+@ nkf-perl
+sdesc: "Network Kanji Filter - perl module"
+ldesc: "This is a Perl Extension version of nkf (Network Kanji Filter). 
+It converts the last argument and return converted result. 
+Conversion details are specified by flags before the last argument.
+This package is built against perl 5.10.0."
+category: Utils
+requires: cygwin perl
