--- origsrc/boost_1_44_0/boost/system/api_config.hpp	2010-07-06 05:47:30.000000000 -0500
+++ src/boost_1_44_0/boost/system/api_config.hpp	2010-09-28 16:10:09.597043200 -0500
@@ -33,7 +33,7 @@
 //    Standalone MinGW and all other known Windows compilers do predefine _WIN32
 //    Compilers that predefine _WIN32 or __MINGW32__ do so for Windows 64-bit builds too.
 
-# if defined(_WIN32) || defined(__CYGWIN__) // Windows default, including MinGW and Cygwin
+# if defined(_WIN32) // Windows default, including MinGW
 #   define BOOST_WINDOWS_API
 # else
 #   define BOOST_POSIX_API 
--- origsrc/boost_1_46_1/libs/system/src/error_code.cpp	2011-01-11 14:49:15.000000000 -0600
+++ src/boost_1_46_1/libs/system/src/error_code.cpp	2011-06-10 02:18:27.750904100 -0500
@@ -94,7 +94,7 @@ namespace
       char buf[64];
       char * bp = buf;
       std::size_t sz = sizeof(buf);
-  #  if defined(__CYGWIN__) || defined(__USE_GNU)
+  #  if defined(__USE_GNU)
       // Oddball version of strerror_r
       const char * c_str = strerror_r( ev, bp, sz );
       return  c_str
