--- origsrc/boost_1_48_0/boost/asio/detail/socket_types.hpp	2012-01-02 03:33:22.350585600 -0600
+++ src/boost_1_48_0/boost/asio/detail/socket_types.hpp	2012-01-02 03:41:35.817810300 -0600
@@ -157,6 +157,9 @@ const int shutdown_both = SHUT_RDWR;
 const int message_peek = MSG_PEEK;
 const int message_out_of_band = MSG_OOB;
 const int message_do_not_route = MSG_DONTROUTE;
+#ifndef MSG_EOR
+#define MSG_EOR 0
+#endif
 const int message_end_of_record = MSG_EOR;
 # if defined(IOV_MAX)
 const int max_iov_len = IOV_MAX;
