--- origsrc/boost_1_53_0/boost/asio/detail/buffer_sequence_adapter.hpp	2013-04-17 17:53:30.950191900 -0500
+++ src/boost_1_53_0/boost/asio/detail/buffer_sequence_adapter.hpp	2013-04-17 17:21:00.000000000 -0500
@@ -29,7 +29,7 @@ namespace detail {
 class buffer_sequence_adapter_base
 {
 protected:
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   typedef WSABUF native_buffer_type;
 
   static void init_native_buffer(WSABUF& buf,
@@ -45,7 +45,7 @@ protected:
     buf.buf = const_cast<char*>(boost::asio::buffer_cast<const char*>(buffer));
     buf.len = static_cast<ULONG>(boost::asio::buffer_size(buffer));
   }
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   typedef iovec native_buffer_type;
 
   static void init_iov_base(void*& base, void* addr)
@@ -73,7 +73,7 @@ protected:
           boost::asio::buffer_cast<const void*>(buffer)));
     iov.iov_len = boost::asio::buffer_size(buffer);
   }
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 };
 
 // Helper class to translate buffers into the native buffer representation.
--- origsrc/boost_1_53_0/boost/asio/detail/config.hpp	2013-04-17 17:53:30.972193100 -0500
+++ src/boost_1_53_0/boost/asio/detail/config.hpp	2013-04-17 17:21:01.000000000 -0500
@@ -195,7 +195,7 @@
 #endif // !defined(BOOST_ASIO_DISABLE_BOOST_CHRONO)
 
 // Windows: target OS version.
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # if !defined(_WIN32_WINNT) && !defined(_WIN32_WINDOWS)
 #  if defined(_MSC_VER) || defined(__BORLANDC__)
 #   pragma message( \
@@ -233,28 +233,28 @@
 #   error You must add -D__USE_W32_SOCKETS to your compiler options.
 #  endif // !defined(__USE_W32_SOCKETS)
 # endif // defined(__CYGWIN__)
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 // Windows: minimise header inclusion.
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # if !defined(BOOST_ASIO_NO_WIN32_LEAN_AND_MEAN)
 #  if !defined(WIN32_LEAN_AND_MEAN)
 #   define WIN32_LEAN_AND_MEAN
 #  endif // !defined(WIN32_LEAN_AND_MEAN)
 # endif // !defined(BOOST_ASIO_NO_WIN32_LEAN_AND_MEAN)
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 // Windows: suppress definition of "min" and "max" macros.
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # if !defined(BOOST_ASIO_NO_NOMINMAX)
 #  if !defined(NOMINMAX)
 #   define NOMINMAX 1
 #  endif // !defined(NOMINMAX)
 # endif // !defined(BOOST_ASIO_NO_NOMINMAX)
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 // Windows: IO Completion Ports.
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # if defined(_WIN32_WINNT) && (_WIN32_WINNT >= 0x0400)
 #  if !defined(UNDER_CE)
 #   if !defined(BOOST_ASIO_DISABLE_IOCP)
@@ -262,7 +262,7 @@
 #   endif // !defined(BOOST_ASIO_DISABLE_IOCP)
 #  endif // !defined(UNDER_CE)
 # endif // defined(_WIN32_WINNT) && (_WIN32_WINNT >= 0x0400)
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 // Linux: epoll, eventfd and timerfd.
 #if defined(__linux__)
@@ -306,14 +306,14 @@
 
 // Serial ports.
 #if defined(BOOST_ASIO_HAS_IOCP) \
-   || !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+   || !defined(BOOST_WINDOWS)
 # if !defined(__SYMBIAN32__)
 #  if !defined(BOOST_ASIO_DISABLE_SERIAL_PORT)
 #   define BOOST_ASIO_HAS_SERIAL_PORT 1
 #  endif // !defined(BOOST_ASIO_DISABLE_SERIAL_PORT)
 # endif // !defined(__SYMBIAN32__)
 #endif // defined(BOOST_ASIO_HAS_IOCP)
-       //   || !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+       //   || !defined(BOOST_WINDOWS)
 
 // Windows: stream handles.
 #if !defined(BOOST_ASIO_DISABLE_WINDOWS_STREAM_HANDLE)
@@ -331,11 +331,11 @@
 
 // Windows: object handles.
 #if !defined(BOOST_ASIO_DISABLE_WINDOWS_OBJECT_HANDLE)
-# if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# if defined(BOOST_WINDOWS)
 #  if !defined(UNDER_CE)
 #   define BOOST_ASIO_HAS_WINDOWS_OBJECT_HANDLE 1
 #  endif // !defined(UNDER_CE)
-# endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# endif // defined(BOOST_WINDOWS)
 #endif // !defined(BOOST_ASIO_DISABLE_WINDOWS_OBJECT_HANDLE)
 
 // Windows: OVERLAPPED wrapper.
@@ -347,23 +347,23 @@
 
 // POSIX: stream-oriented file descriptors.
 #if !defined(BOOST_ASIO_DISABLE_POSIX_STREAM_DESCRIPTOR)
-# if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+# if !defined(BOOST_WINDOWS)
 #  define BOOST_ASIO_HAS_POSIX_STREAM_DESCRIPTOR 1
-# endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+# endif // !defined(BOOST_WINDOWS)
 #endif // !defined(BOOST_ASIO_DISABLE_POSIX_STREAM_DESCRIPTOR)
 
 // UNIX domain sockets.
 #if !defined(BOOST_ASIO_DISABLE_LOCAL_SOCKETS)
-# if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+# if !defined(BOOST_WINDOWS)
 #  define BOOST_ASIO_HAS_LOCAL_SOCKETS 1
-# endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+# endif // !defined(BOOST_WINDOWS)
 #endif // !defined(BOOST_ASIO_DISABLE_LOCAL_SOCKETS)
 
 // Can use sigaction() instead of signal().
 #if !defined(BOOST_ASIO_DISABLE_SIGACTION)
-# if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+# if !defined(BOOST_WINDOWS)
 #  define BOOST_ASIO_HAS_SIGACTION 1
-# endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+# endif // !defined(BOOST_WINDOWS)
 #endif // !defined(BOOST_ASIO_DISABLE_SIGACTION)
 
 // Can use signal().
--- origsrc/boost_1_53_0/boost/asio/detail/descriptor_ops.hpp	2013-04-17 17:53:30.989194100 -0500
+++ src/boost_1_53_0/boost/asio/detail/descriptor_ops.hpp	2013-04-17 17:21:01.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 #include <cstddef>
 #include <boost/system/error_code.hpp>
@@ -110,6 +110,6 @@ BOOST_ASIO_DECL int poll_write(int d,
 # include <boost/asio/detail/impl/descriptor_ops.ipp>
 #endif // defined(BOOST_ASIO_HEADER_ONLY)
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_DESCRIPTOR_OPS_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/descriptor_read_op.hpp	2013-04-17 17:53:31.007195100 -0500
+++ src/boost_1_53_0/boost/asio/detail/descriptor_read_op.hpp	2013-04-17 17:21:01.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 #include <boost/utility/addressof.hpp>
 #include <boost/asio/detail/bind_handler.hpp>
@@ -116,6 +116,6 @@ private:
 
 #include <boost/asio/detail/pop_options.hpp>
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_DESCRIPTOR_READ_OP_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/descriptor_write_op.hpp	2013-04-17 17:53:31.024196100 -0500
+++ src/boost_1_53_0/boost/asio/detail/descriptor_write_op.hpp	2013-04-17 17:21:01.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 #include <boost/utility/addressof.hpp>
 #include <boost/asio/detail/bind_handler.hpp>
@@ -116,6 +116,6 @@ private:
 
 #include <boost/asio/detail/pop_options.hpp>
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_DESCRIPTOR_WRITE_OP_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/fd_set_adapter.hpp	2013-04-17 17:53:31.041197100 -0500
+++ src/boost_1_53_0/boost/asio/detail/fd_set_adapter.hpp	2013-04-17 17:21:01.000000000 -0500
@@ -23,7 +23,7 @@ namespace boost {
 namespace asio {
 namespace detail {
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 typedef win_fd_set_adapter fd_set_adapter;
 #else
 typedef posix_fd_set_adapter fd_set_adapter;
--- origsrc/boost_1_53_0/boost/asio/detail/hash_map.hpp	2013-04-17 17:53:31.059198100 -0500
+++ src/boost_1_53_0/boost/asio/detail/hash_map.hpp	2013-04-17 17:21:02.000000000 -0500
@@ -21,9 +21,9 @@
 #include <utility>
 #include <boost/asio/detail/noncopyable.hpp>
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # include <boost/asio/detail/socket_types.hpp>
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 #include <boost/asio/detail/push_options.hpp>
 
@@ -42,12 +42,12 @@ inline std::size_t calculate_hash_value(
     + (reinterpret_cast<std::size_t>(p) >> 3);
 }
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 inline std::size_t calculate_hash_value(SOCKET s)
 {
   return static_cast<std::size_t>(s);
 }
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 // Note: assumes K and V are POD types.
 template <typename K, typename V>
--- origsrc/boost_1_53_0/boost/asio/detail/impl/descriptor_ops.ipp	2013-04-17 17:53:31.078199200 -0500
+++ src/boost_1_53_0/boost/asio/detail/impl/descriptor_ops.ipp	2013-04-17 17:21:02.000000000 -0500
@@ -20,7 +20,7 @@
 #include <boost/asio/detail/descriptor_ops.hpp>
 #include <boost/asio/error.hpp>
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 #include <boost/asio/detail/push_options.hpp>
 
@@ -444,6 +444,6 @@ int poll_write(int d, state_type state,
 
 #include <boost/asio/detail/pop_options.hpp>
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_IMPL_DESCRIPTOR_OPS_IPP
--- origsrc/boost_1_53_0/boost/asio/detail/impl/pipe_select_interrupter.ipp	2013-04-17 17:53:31.099200400 -0500
+++ src/boost_1_53_0/boost/asio/detail/impl/pipe_select_interrupter.ipp	2013-04-17 17:21:02.000000000 -0500
@@ -18,7 +18,6 @@
 #include <boost/asio/detail/config.hpp>
 
 #if !defined(BOOST_WINDOWS)
-#if !defined(__CYGWIN__)
 #if !defined(__SYMBIAN32__)
 #if !defined(BOOST_ASIO_HAS_EVENTFD)
 
@@ -118,7 +117,6 @@ bool pipe_select_interrupter::reset()
 
 #endif // !defined(BOOST_ASIO_HAS_EVENTFD)
 #endif // !defined(__SYMBIAN32__)
-#endif // !defined(__CYGWIN__)
 #endif // !defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_IMPL_PIPE_SELECT_INTERRUPTER_IPP
--- origsrc/boost_1_53_0/boost/asio/detail/impl/reactive_descriptor_service.ipp	2013-04-17 17:53:31.114201200 -0500
+++ src/boost_1_53_0/boost/asio/detail/impl/reactive_descriptor_service.ipp	2013-04-17 17:21:02.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 #include <boost/asio/error.hpp>
 #include <boost/asio/detail/reactive_descriptor_service.hpp>
@@ -200,6 +200,6 @@ void reactive_descriptor_service::start_
 
 #include <boost/asio/detail/pop_options.hpp>
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_IMPL_REACTIVE_DESCRIPTOR_SERVICE_IPP
--- origsrc/boost_1_53_0/boost/asio/detail/impl/reactive_serial_port_service.ipp	2013-04-17 17:53:31.120201600 -0500
+++ src/boost_1_53_0/boost/asio/detail/impl/reactive_serial_port_service.ipp	2013-04-17 17:21:02.000000000 -0500
@@ -19,7 +19,7 @@
 #include <boost/asio/detail/config.hpp>
 
 #if defined(BOOST_ASIO_HAS_SERIAL_PORT)
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 #include <cstring>
 #include <boost/asio/detail/reactive_serial_port_service.hpp>
@@ -147,7 +147,7 @@ boost::system::error_code reactive_seria
 
 #include <boost/asio/detail/pop_options.hpp>
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 #endif // defined(BOOST_ASIO_HAS_SERIAL_PORT)
 
 #endif // BOOST_ASIO_DETAIL_IMPL_REACTIVE_SERIAL_PORT_SERVICE_IPP
--- origsrc/boost_1_53_0/boost/asio/detail/impl/select_reactor.ipp	2013-04-17 17:53:31.126201900 -0500
+++ src/boost_1_53_0/boost/asio/detail/impl/select_reactor.ipp	2013-04-17 17:21:02.000000000 -0500
@@ -185,7 +185,7 @@ void select_reactor::run(bool block, op_
       max_fd = fd_sets_[i].max_descriptor();
   }
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   // Connection operations on Windows use both except and write fd_sets.
   have_work_to_do = have_work_to_do || !op_queue_[connect_op].empty();
   op_queue_[connect_op].get_descriptors(fd_sets_[write_op], ops);
@@ -194,7 +194,7 @@ void select_reactor::run(bool block, op_
   op_queue_[connect_op].get_descriptors(fd_sets_[except_op], ops);
   if (fd_sets_[except_op].max_descriptor() > max_fd)
     max_fd = fd_sets_[except_op].max_descriptor();
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
   // We can return immediately if there's no work to do and the reactor is
   // not supposed to block.
@@ -224,13 +224,13 @@ void select_reactor::run(bool block, op_
   // Dispatch all ready operations.
   if (retval > 0)
   {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
     // Connection operations on Windows use both except and write fd_sets.
     op_queue_[connect_op].perform_operations_for_descriptors(
         fd_sets_[except_op], ops);
     op_queue_[connect_op].perform_operations_for_descriptors(
         fd_sets_[write_op], ops);
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
     // Exception operations must be processed first to ensure that any
     // out-of-band data is read before normal data.
--- origsrc/boost_1_53_0/boost/asio/detail/impl/signal_set_service.ipp	2012-12-30 17:17:13.000000000 -0600
+++ src/boost_1_53_0/boost/asio/detail/impl/signal_set_service.ipp	2013-04-17 17:56:52.740733600 -0500
@@ -59,23 +59,23 @@ signal_state* get_signal_state()
 
 void asio_signal_handler(int signal_number)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   signal_set_service::deliver_signal(signal_number);
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   int saved_errno = errno;
   signal_state* state = get_signal_state();
   signed_size_type result = ::write(state->write_descriptor_,
       &signal_number, sizeof(signal_number));
   (void)result;
   errno = saved_errno;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 #if defined(BOOST_ASIO_HAS_SIGNAL) && !defined(BOOST_ASIO_HAS_SIGACTION)
   ::signal(signal_number, asio_signal_handler);
 #endif // defined(BOOST_ASIO_HAS_SIGNAL) && !defined(BOOST_ASIO_HAS_SIGACTION)
 }
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 class signal_set_service::pipe_read_op : public reactor_op
 {
 public:
@@ -105,22 +105,22 @@ public:
     delete o;
   }
 };
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 signal_set_service::signal_set_service(
     boost::asio::io_service& io_service)
   : io_service_(boost::asio::use_service<io_service_impl>(io_service)),
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
     reactor_(boost::asio::use_service<reactor>(io_service)),
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
     next_(0),
     prev_(0)
 {
   get_signal_state()->mutex_.init();
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
   reactor_.init_task();
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
   for (int i = 0; i < max_signal_number; ++i)
     registrations_[i] = 0;
@@ -155,7 +155,7 @@ void signal_set_service::shutdown_servic
 void signal_set_service::fork_service(
     boost::asio::io_service::fork_event fork_ev)
 {
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
   signal_state* state = get_signal_state();
   static_mutex::scoped_lock lock(state->mutex_);
 
@@ -185,9 +185,9 @@ void signal_set_service::fork_service(
   default:
     break;
   }
-#else // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#else // !defined(BOOST_WINDOWS)
   (void)fork_ev;
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 }
 
 void signal_set_service::construct(
@@ -247,12 +247,12 @@ boost::system::error_code signal_set_ser
       if (::signal(signal_number, asio_signal_handler) == SIG_ERR)
 # endif // defined(BOOST_ASIO_HAS_SIGACTION)
       {
-# if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# if defined(BOOST_WINDOWS)
         ec = boost::asio::error::invalid_argument;
-# else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# else // defined(BOOST_WINDOWS)
         ec = boost::system::error_code(errno,
             boost::asio::error::get_system_category());
-# endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# endif // defined(BOOST_WINDOWS)
         delete new_registration;
         return ec;
       }
@@ -317,12 +317,12 @@ boost::system::error_code signal_set_ser
       if (::signal(signal_number, SIG_DFL) == SIG_ERR)
 # endif // defined(BOOST_ASIO_HAS_SIGACTION)
       {
-# if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# if defined(BOOST_WINDOWS)
         ec = boost::asio::error::invalid_argument;
-# else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# else // defined(BOOST_WINDOWS)
         ec = boost::system::error_code(errno,
             boost::asio::error::get_system_category());
-# endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# endif // defined(BOOST_WINDOWS)
         return ec;
       }
     }
@@ -371,12 +371,12 @@ boost::system::error_code signal_set_ser
       if (::signal(reg->signal_number_, SIG_DFL) == SIG_ERR)
 # endif // defined(BOOST_ASIO_HAS_SIGACTION)
       {
-# if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# if defined(BOOST_WINDOWS)
         ec = boost::asio::error::invalid_argument;
-# else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# else // defined(BOOST_WINDOWS)
         ec = boost::system::error_code(errno,
             boost::asio::error::get_system_category());
-# endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# endif // defined(BOOST_WINDOWS)
         return ec;
       }
     }
@@ -466,11 +466,11 @@ void signal_set_service::add_service(sig
   signal_state* state = get_signal_state();
   static_mutex::scoped_lock lock(state->mutex_);
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
   // If this is the first service to be created, open a new pipe.
   if (state->service_list_ == 0)
     open_descriptors();
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
   // Insert service into linked list of all services.
   service->next_ = state->service_list_;
@@ -479,11 +479,11 @@ void signal_set_service::add_service(sig
     state->service_list_->prev_ = service;
   state->service_list_ = service;
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
   // Register for pipe readiness notifications.
   service->reactor_.register_internal_descriptor(reactor::read_op,
       state->read_descriptor_, service->reactor_data_, new pipe_read_op);
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 }
 
 void signal_set_service::remove_service(signal_set_service* service)
@@ -493,11 +493,11 @@ void signal_set_service::remove_service(
 
   if (service->next_ || service->prev_ || state->service_list_ == service)
   {
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
     // Disable the pipe readiness notifications.
     service->reactor_.deregister_descriptor(
         state->read_descriptor_, service->reactor_data_, false);
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
     // Remove service from linked list of all services.
     if (state->service_list_ == service)
@@ -509,17 +509,17 @@ void signal_set_service::remove_service(
     service->next_ = 0;
     service->prev_ = 0;
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
     // If this is the last service to be removed, close the pipe.
     if (state->service_list_ == 0)
       close_descriptors();
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
   }
 }
 
 void signal_set_service::open_descriptors()
 {
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
   signal_state* state = get_signal_state();
 
   int pipe_fds[2];
@@ -542,12 +542,12 @@ void signal_set_service::open_descriptor
         boost::asio::error::get_system_category());
     boost::asio::detail::throw_error(ec, "signal_set_service pipe");
   }
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 }
 
 void signal_set_service::close_descriptors()
 {
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
   signal_state* state = get_signal_state();
 
   if (state->read_descriptor_ != -1)
@@ -557,7 +557,7 @@ void signal_set_service::close_descripto
   if (state->write_descriptor_ != -1)
     ::close(state->write_descriptor_);
   state->write_descriptor_ = -1;
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 }
 
 void signal_set_service::start_wait_op(
--- origsrc/boost_1_53_0/boost/asio/detail/impl/socket_ops.ipp	2012-12-30 17:17:13.000000000 -0600
+++ src/boost_1_53_0/boost/asio/detail/impl/socket_ops.ipp	2013-04-17 17:56:03.237902200 -0500
@@ -34,9 +34,9 @@ namespace asio {
 namespace detail {
 namespace socket_ops {
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 struct msghdr { int msg_namelen; };
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 #if defined(__hpux)
 // HP-UX doesn't declare these functions extern "C", so they are declared again
@@ -47,7 +47,7 @@ extern "C" unsigned int if_nametoindex(c
 
 inline void clear_last_error()
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   WSASetLastError(0);
 #else
   errno = 0;
@@ -58,7 +58,7 @@ template <typename ReturnType>
 inline ReturnType error_wrapper(ReturnType return_value,
     boost::system::error_code& ec)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   ec = boost::system::error_code(WSAGetLastError(),
       boost::asio::error::get_system_category());
 #else
@@ -293,11 +293,11 @@ int close(socket_type s, state_type& sta
     }
 
     clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
     result = error_wrapper(::closesocket(s), ec);
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
     result = error_wrapper(::close(s), ec);
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
     if (result != 0
         && (ec == boost::asio::error::would_block
@@ -309,10 +309,10 @@ int close(socket_type s, state_type& sta
       // current OS where this behaviour is seen, Windows, says that the socket
       // remains open. Therefore we'll put the descriptor back into blocking
       // mode and have another attempt at closing it.
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
       ioctl_arg_type arg = 0;
       ::ioctlsocket(s, FIONBIO, &arg);
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
 # if defined(__SYMBIAN32__)
       int flags = ::fcntl(s, F_GETFL, 0);
       if (flags >= 0)
@@ -321,15 +321,15 @@ int close(socket_type s, state_type& sta
       ioctl_arg_type arg = 0;
       ::ioctl(s, FIONBIO, &arg);
 # endif // defined(__SYMBIAN32__)
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
       state &= ~non_blocking;
 
       clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
       result = error_wrapper(::closesocket(s), ec);
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
       result = error_wrapper(::close(s), ec);
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
     }
   }
 
@@ -348,7 +348,7 @@ bool set_user_non_blocking(socket_type s
   }
 
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   ioctl_arg_type arg = (value ? 1 : 0);
   int result = error_wrapper(::ioctlsocket(s, FIONBIO, &arg), ec);
 #elif defined(__SYMBIAN32__)
@@ -401,7 +401,7 @@ bool set_internal_non_blocking(socket_ty
   }
 
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   ioctl_arg_type arg = (value ? 1 : 0);
   int result = error_wrapper(::ioctlsocket(s, FIONBIO, &arg), ec);
 #elif defined(__SYMBIAN32__)
@@ -524,7 +524,7 @@ bool non_blocking_connect(socket_type s,
 int socketpair(int af, int type, int protocol,
     socket_type sv[2], boost::system::error_code& ec)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   (void)(af);
   (void)(type);
   (void)(protocol);
@@ -550,11 +550,11 @@ bool sockatmark(socket_type s, boost::sy
 
 #if defined(SIOCATMARK)
   ioctl_arg_type value = 0;
-# if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# if defined(BOOST_WINDOWS)
   int result = error_wrapper(::ioctlsocket(s, SIOCATMARK, &value), ec);
-# else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# else // defined(BOOST_WINDOWS)
   int result = error_wrapper(::ioctl(s, SIOCATMARK, &value), ec);
-# endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+# endif // defined(BOOST_WINDOWS)
   if (result == 0)
     ec = boost::system::error_code();
 # if defined(ENOTTY)
@@ -579,11 +579,11 @@ size_t available(socket_type s, boost::s
   }
 
   ioctl_arg_type value = 0;
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   int result = error_wrapper(::ioctlsocket(s, FIONREAD, &value), ec);
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   int result = error_wrapper(::ioctl(s, FIONREAD, &value), ec);
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
   if (result == 0)
     ec = boost::system::error_code();
 #if defined(ENOTTY)
@@ -620,32 +620,32 @@ inline void init_buf_iov_base(T& base, v
   base = static_cast<T>(addr);
 }
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 typedef WSABUF buf;
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
 typedef iovec buf;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 void init_buf(buf& b, void* data, size_t size)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   b.buf = static_cast<char*>(data);
   b.len = static_cast<u_long>(size);
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   init_buf_iov_base(b.iov_base, data);
   b.iov_len = size;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 }
 
 void init_buf(buf& b, const void* data, size_t size)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   b.buf = static_cast<char*>(const_cast<void*>(data));
   b.len = static_cast<u_long>(size);
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   init_buf_iov_base(b.iov_base, const_cast<void*>(data));
   b.iov_len = size;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 }
 
 inline void init_msghdr_msg_name(void*& name, socket_addr_type* addr)
@@ -674,7 +674,7 @@ signed_size_type recv(socket_type s, buf
     int flags, boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   // Receive some data.
   DWORD recv_buf_count = static_cast<DWORD>(count);
   DWORD bytes_transferred = 0;
@@ -689,7 +689,7 @@ signed_size_type recv(socket_type s, buf
     return socket_error_retval;
   ec = boost::system::error_code();
   return bytes_transferred;
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   msghdr msg = msghdr();
   msg.msg_iov = bufs;
   msg.msg_iovlen = static_cast<int>(count);
@@ -697,7 +697,7 @@ signed_size_type recv(socket_type s, buf
   if (result >= 0)
     ec = boost::system::error_code();
   return result;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 }
 
 size_t sync_recv(socket_type s, state_type state, buf* bufs,
@@ -820,7 +820,7 @@ signed_size_type recvfrom(socket_type s,
     boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   // Receive some data.
   DWORD recv_buf_count = static_cast<DWORD>(count);
   DWORD bytes_transferred = 0;
@@ -837,7 +837,7 @@ signed_size_type recvfrom(socket_type s,
     return socket_error_retval;
   ec = boost::system::error_code();
   return bytes_transferred;
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   msghdr msg = msghdr();
   init_msghdr_msg_name(msg.msg_name, addr);
   msg.msg_namelen = static_cast<int>(*addrlen);
@@ -848,7 +848,7 @@ signed_size_type recvfrom(socket_type s,
   if (result >= 0)
     ec = boost::system::error_code();
   return result;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 }
 
 size_t sync_recvfrom(socket_type s, state_type state, buf* bufs,
@@ -945,10 +945,10 @@ signed_size_type recvmsg(socket_type s,
     int in_flags, int& out_flags, boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   out_flags = 0;
   return socket_ops::recv(s, bufs, count, in_flags, ec);
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   msghdr msg = msghdr();
   msg.msg_iov = bufs;
   msg.msg_iovlen = static_cast<int>(count);
@@ -961,7 +961,7 @@ signed_size_type recvmsg(socket_type s,
   else
     out_flags = 0;
   return result;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 }
 
 size_t sync_recvmsg(socket_type s, state_type state,
@@ -1057,7 +1057,7 @@ signed_size_type send(socket_type s, con
     int flags, boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   // Send the data.
   DWORD send_buf_count = static_cast<DWORD>(count);
   DWORD bytes_transferred = 0;
@@ -1072,7 +1072,7 @@ signed_size_type send(socket_type s, con
     return socket_error_retval;
   ec = boost::system::error_code();
   return bytes_transferred;
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   msghdr msg = msghdr();
   msg.msg_iov = const_cast<buf*>(bufs);
   msg.msg_iovlen = static_cast<int>(count);
@@ -1083,7 +1083,7 @@ signed_size_type send(socket_type s, con
   if (result >= 0)
     ec = boost::system::error_code();
   return result;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 }
 
 size_t sync_send(socket_type s, state_type state, const buf* bufs,
@@ -1184,7 +1184,7 @@ signed_size_type sendto(socket_type s, c
     boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   // Send the data.
   DWORD send_buf_count = static_cast<DWORD>(count);
   DWORD bytes_transferred = 0;
@@ -1199,7 +1199,7 @@ signed_size_type sendto(socket_type s, c
     return socket_error_retval;
   ec = boost::system::error_code();
   return bytes_transferred;
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   msghdr msg = msghdr();
   init_msghdr_msg_name(msg.msg_name, addr);
   msg.msg_namelen = static_cast<int>(addrlen);
@@ -1212,7 +1212,7 @@ signed_size_type sendto(socket_type s, c
   if (result >= 0)
     ec = boost::system::error_code();
   return result;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 }
 
 size_t sync_sendto(socket_type s, state_type state, const buf* bufs,
@@ -1289,7 +1289,7 @@ socket_type socket(int af, int type, int
     boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   socket_type s = error_wrapper(::WSASocket(af, type, protocol, 0, 0,
         WSA_FLAG_OVERLAPPED), ec);
   if (s == invalid_socket)
@@ -1487,7 +1487,7 @@ int getsockopt(socket_type s, state_type
   }
   ec = boost::asio::error::fault;
   return socket_error_retval;
-#elif defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#elif defined(BOOST_WINDOWS)
   clear_last_error();
   int result = error_wrapper(call_getsockopt(&msghdr::msg_namelen,
         s, level, optname, optval, optlen), ec);
@@ -1505,7 +1505,7 @@ int getsockopt(socket_type s, state_type
   if (result == 0)
     ec = boost::system::error_code();
   return result;
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   clear_last_error();
   int result = error_wrapper(call_getsockopt(&msghdr::msg_namelen,
         s, level, optname, optval, optlen), ec);
@@ -1524,7 +1524,7 @@ int getsockopt(socket_type s, state_type
   if (result == 0)
     ec = boost::system::error_code();
   return result;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 }
 
 template <typename SockLenType>
@@ -1546,7 +1546,7 @@ int getpeername(socket_type s, socket_ad
     return socket_error_retval;
   }
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   if (cached)
   {
     // Check if socket is still connected.
@@ -1567,9 +1567,9 @@ int getpeername(socket_type s, socket_ad
     ec = boost::system::error_code();
     return 0;
   }
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   (void)cached;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
   clear_last_error();
   int result = error_wrapper(call_getpeername(
@@ -1616,7 +1616,7 @@ int ioctl(socket_type s, state_type& sta
   }
 
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   int result = error_wrapper(::ioctlsocket(s, cmd, arg), ec);
 #elif defined(__MACH__) && defined(__APPLE__) \
   || defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__)
@@ -1657,7 +1657,7 @@ int select(int nfds, fd_set* readfds, fd
     fd_set* exceptfds, timeval* timeout, boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   if (!readfds && !writefds && !exceptfds && timeout)
   {
     DWORD milliseconds = timeout->tv_sec * 1000 + timeout->tv_usec / 1000;
@@ -1677,7 +1677,7 @@ int select(int nfds, fd_set* readfds, fd
   if (timeout && timeout->tv_sec == 0
       && timeout->tv_usec > 0 && timeout->tv_usec < 1000)
     timeout->tv_usec = 1000;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 #if defined(__hpux) && defined(__SELECT)
   timespec ts;
@@ -1703,7 +1703,6 @@ int poll_read(socket_type s, state_type
   }
 
 #if defined(BOOST_WINDOWS) \
-  || defined(__CYGWIN__) \
   || defined(__SYMBIAN32__)
   fd_set fds;
   FD_ZERO(&fds);
@@ -1715,7 +1714,6 @@ int poll_read(socket_type s, state_type
   clear_last_error();
   int result = error_wrapper(::select(s, &fds, 0, 0, timeout), ec);
 #else // defined(BOOST_WINDOWS)
-      // || defined(__CYGWIN__)
       // || defined(__SYMBIAN32__)
   pollfd fds;
   fds.fd = s;
@@ -1725,7 +1723,6 @@ int poll_read(socket_type s, state_type
   clear_last_error();
   int result = error_wrapper(::poll(&fds, 1, timeout), ec);
 #endif // defined(BOOST_WINDOWS)
-       // || defined(__CYGWIN__)
        // || defined(__SYMBIAN32__)
   if (result == 0)
     ec = (state & user_set_non_blocking)
@@ -1744,7 +1741,6 @@ int poll_write(socket_type s, state_type
   }
 
 #if defined(BOOST_WINDOWS) \
-  || defined(__CYGWIN__) \
   || defined(__SYMBIAN32__)
   fd_set fds;
   FD_ZERO(&fds);
@@ -1756,7 +1752,6 @@ int poll_write(socket_type s, state_type
   clear_last_error();
   int result = error_wrapper(::select(s, 0, &fds, 0, timeout), ec);
 #else // defined(BOOST_WINDOWS)
-      // || defined(__CYGWIN__)
       // || defined(__SYMBIAN32__)
   pollfd fds;
   fds.fd = s;
@@ -1766,7 +1761,6 @@ int poll_write(socket_type s, state_type
   clear_last_error();
   int result = error_wrapper(::poll(&fds, 1, timeout), ec);
 #endif // defined(BOOST_WINDOWS)
-       // || defined(__CYGWIN__)
        // || defined(__SYMBIAN32__)
   if (result == 0)
     ec = (state & user_set_non_blocking)
@@ -1785,7 +1779,6 @@ int poll_connect(socket_type s, boost::s
   }
 
 #if defined(BOOST_WINDOWS) \
-  || defined(__CYGWIN__) \
   || defined(__SYMBIAN32__)
   fd_set write_fds;
   FD_ZERO(&write_fds);
@@ -1799,7 +1792,6 @@ int poll_connect(socket_type s, boost::s
     ec = boost::system::error_code();
   return result;
 #else // defined(BOOST_WINDOWS)
-      // || defined(__CYGWIN__)
       // || defined(__SYMBIAN32__)
   pollfd fds;
   fds.fd = s;
@@ -1811,7 +1803,6 @@ int poll_connect(socket_type s, boost::s
     ec = boost::system::error_code();
   return result;
 #endif // defined(BOOST_WINDOWS)
-       // || defined(__CYGWIN__)
        // || defined(__SYMBIAN32__)
 }
 
@@ -1819,7 +1810,7 @@ const char* inet_ntop(int af, const void
     unsigned long scope_id, boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   using namespace std; // For memcpy.
 
   if (af != AF_INET && af != AF_INET6)
@@ -1873,7 +1864,7 @@ const char* inet_ntop(int af, const void
     ec = boost::asio::error::invalid_argument;
 
   return result == socket_error_retval ? 0 : dest;
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   const char* result = error_wrapper(::inet_ntop(
         af, src, dest, static_cast<int>(length)), ec);
   if (result == 0 && !ec)
@@ -1891,14 +1882,14 @@ const char* inet_ntop(int af, const void
     strcat(dest, if_name);
   }
   return result;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 }
 
 int inet_pton(int af, const char* src, void* dest,
     unsigned long* scope_id, boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   using namespace std; // For memcpy and strcmp.
 
   if (af != AF_INET && af != AF_INET6)
@@ -1958,7 +1949,7 @@ int inet_pton(int af, const char* src, v
     ec = boost::system::error_code();
 
   return result == socket_error_retval ? -1 : 1;
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   int result = error_wrapper(::inet_pton(af, src, dest), ec);
   if (result <= 0 && !ec)
     ec = boost::asio::error::invalid_argument;
@@ -1978,7 +1969,7 @@ int inet_pton(int af, const char* src, v
     }
   }
   return result;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 }
 
 int gethostname(char* name, int namelen, boost::system::error_code& ec)
@@ -1992,7 +1983,7 @@ int gethostname(char* name, int namelen,
   return result;
 }
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__) \
+#if defined(BOOST_WINDOWS) \
   || defined(__MACH__) && defined(__APPLE__)
 
 // The following functions are only needed for emulation of getaddrinfo and
@@ -2022,7 +2013,7 @@ inline hostent* gethostbyaddr(const char
     hostent* result, char* buffer, int buflength, boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   (void)(buffer);
   (void)(buflength);
   hostent* retval = error_wrapper(::gethostbyaddr(addr, length, af), ec);
@@ -2065,7 +2056,7 @@ inline hostent* gethostbyname(const char
     char* buffer, int buflength, int ai_flags, boost::system::error_code& ec)
 {
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   (void)(buffer);
   (void)(buflength);
   (void)(ai_flags);
@@ -2832,7 +2823,7 @@ inline boost::system::error_code getname
   return ec;
 }
 
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
        //   || defined(__MACH__) && defined(__APPLE__)
 
 inline boost::system::error_code translate_addrinfo_error(int error)
@@ -2864,7 +2855,7 @@ inline boost::system::error_code transla
   case EAI_SOCKTYPE:
     return boost::asio::error::socket_type_not_supported;
   default: // Possibly the non-portable EAI_SYSTEM.
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
     return boost::system::error_code(
         WSAGetLastError(), boost::asio::error::get_system_category());
 #else
@@ -2881,7 +2872,7 @@ boost::system::error_code getaddrinfo(co
   host = (host && *host) ? host : 0;
   service = (service && *service) ? service : 0;
   clear_last_error();
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # if defined(_WIN32_WINNT) && (_WIN32_WINNT >= 0x0501) || defined(UNDER_CE)
   // Building for Windows XP, Windows Server 2003, or later.
   int error = ::getaddrinfo(host, service, &hints, result);
@@ -2924,7 +2915,7 @@ boost::system::error_code background_get
 
 void freeaddrinfo(addrinfo_type* ai)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # if defined(_WIN32_WINNT) && (_WIN32_WINNT >= 0x0501) || defined(UNDER_CE)
   // Building for Windows XP, Windows Server 2003, or later.
   ::freeaddrinfo(ai);
@@ -2952,7 +2943,7 @@ boost::system::error_code getnameinfo(co
     std::size_t addrlen, char* host, std::size_t hostlen,
     char* serv, std::size_t servlen, int flags, boost::system::error_code& ec)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # if defined(_WIN32_WINNT) && (_WIN32_WINNT >= 0x0501) || defined(UNDER_CE)
   // Building for Windows XP, Windows Server 2003, or later.
   clear_last_error();
--- origsrc/boost_1_53_0/boost/asio/detail/impl/socket_select_interrupter.ipp	2013-04-17 17:53:31.131202200 -0500
+++ src/boost_1_53_0/boost/asio/detail/impl/socket_select_interrupter.ipp	2013-04-17 17:21:02.000000000 -0500
@@ -18,7 +18,7 @@
 #include <boost/asio/detail/config.hpp>
 
 #if defined(BOOST_WINDOWS) \
-  || defined(__CYGWIN__) \
+  \
   || defined(__SYMBIAN32__)
 
 #include <cstdlib>
@@ -167,7 +167,7 @@ bool socket_select_interrupter::reset()
 #include <boost/asio/detail/pop_options.hpp>
 
 #endif // defined(BOOST_WINDOWS)
-       // || defined(__CYGWIN__)
+       //
        // || defined(__SYMBIAN32__)
 
 #endif // BOOST_ASIO_DETAIL_IMPL_SOCKET_SELECT_INTERRUPTER_IPP
--- origsrc/boost_1_53_0/boost/asio/detail/impl/winsock_init.ipp	2013-04-17 17:53:31.137202600 -0500
+++ src/boost_1_53_0/boost/asio/detail/impl/winsock_init.ipp	2013-04-17 17:21:02.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 
 #include <boost/asio/detail/socket_types.hpp>
 #include <boost/asio/detail/winsock_init.hpp>
@@ -66,6 +66,6 @@ void winsock_init_base::throw_on_error(d
 
 #include <boost/asio/detail/pop_options.hpp>
 
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_IMPL_WINSOCK_INIT_IPP
--- origsrc/boost_1_53_0/boost/asio/detail/local_free_on_block_exit.hpp	2013-04-17 17:53:31.142202800 -0500
+++ src/boost_1_53_0/boost/asio/detail/local_free_on_block_exit.hpp	2013-04-17 17:21:02.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 
 #include <boost/asio/detail/noncopyable.hpp>
 #include <boost/asio/detail/socket_types.hpp>
@@ -54,6 +54,6 @@ private:
 
 #include <boost/asio/detail/pop_options.hpp>
 
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_LOCAL_FREE_ON_BLOCK_EXIT_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/null_signal_blocker.hpp	2013-04-17 17:53:31.147203100 -0500
+++ src/boost_1_53_0/boost/asio/detail/null_signal_blocker.hpp	2013-04-17 17:21:02.000000000 -0500
@@ -20,7 +20,7 @@
 #if !defined(BOOST_HAS_THREADS) \
   || defined(BOOST_ASIO_DISABLE_THREADS) \
   || defined(BOOST_WINDOWS) \
-  || defined(__CYGWIN__) \
+  \
   || defined(__SYMBIAN32__)
 
 #include <boost/asio/detail/noncopyable.hpp>
@@ -65,7 +65,7 @@ public:
 #endif // !defined(BOOST_HAS_THREADS)
        // || defined(BOOST_ASIO_DISABLE_THREADS)
        // || defined(BOOST_WINDOWS)
-       // || defined(__CYGWIN__)
+       //
        // || defined(__SYMBIAN32__)
 
 #endif // BOOST_ASIO_DETAIL_NULL_SIGNAL_BLOCKER_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/old_win_sdk_compat.hpp	2013-04-17 17:53:31.152203400 -0500
+++ src/boost_1_53_0/boost/asio/detail/old_win_sdk_compat.hpp	2013-04-17 17:21:02.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 
 // Guess whether we are building against on old Platform SDK.
 #if !defined(IN6ADDR_ANY_INIT)
@@ -211,6 +211,6 @@ struct addrinfo_emulation
 # define IPPROTO_ICMPV6 58
 #endif
 
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_OLD_WIN_SDK_COMPAT_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/pipe_select_interrupter.hpp	2013-04-17 17:53:31.157203700 -0500
+++ src/boost_1_53_0/boost/asio/detail/pipe_select_interrupter.hpp	2013-04-17 17:21:02.000000000 -0500
@@ -18,7 +18,6 @@
 #include <boost/asio/detail/config.hpp>
 
 #if !defined(BOOST_WINDOWS)
-#if !defined(__CYGWIN__)
 #if !defined(__SYMBIAN32__)
 #if !defined(BOOST_ASIO_HAS_EVENTFD)
 
@@ -83,7 +82,6 @@ private:
 
 #endif // !defined(BOOST_ASIO_HAS_EVENTFD)
 #endif // !defined(__SYMBIAN32__)
-#endif // !defined(__CYGWIN__)
 #endif // !defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_PIPE_SELECT_INTERRUPTER_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/pop_options.hpp	2013-04-17 17:53:31.162204000 -0500
+++ src/boost_1_53_0/boost/asio/detail/pop_options.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -27,7 +27,7 @@
 
 // GNU C++
 
-# if defined(__MINGW32__) || defined(__CYGWIN__)
+# if defined(__MINGW32__)
 #  pragma pack (pop)
 # endif
 
--- origsrc/boost_1_53_0/boost/asio/detail/posix_fd_set_adapter.hpp	2013-04-17 17:53:31.167204300 -0500
+++ src/boost_1_53_0/boost/asio/detail/posix_fd_set_adapter.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 #include <cstring>
 #include <boost/asio/detail/noncopyable.hpp>
@@ -84,6 +84,6 @@ private:
 
 #include <boost/asio/detail/pop_options.hpp>
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_POSIX_FD_SET_ADAPTER_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/push_options.hpp	2013-04-17 17:53:31.173204600 -0500
+++ src/boost_1_53_0/boost/asio/detail/push_options.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -27,7 +27,7 @@
 
 // GNU C++
 
-# if defined(__MINGW32__) || defined(__CYGWIN__)
+# if defined(__MINGW32__)
 #  pragma pack (push, 8)
 # endif
 
--- origsrc/boost_1_53_0/boost/asio/detail/reactive_descriptor_service.hpp	2013-04-17 17:53:31.178204900 -0500
+++ src/boost_1_53_0/boost/asio/detail/reactive_descriptor_service.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 #include <boost/utility/addressof.hpp>
 #include <boost/asio/buffer.hpp>
@@ -303,6 +303,6 @@ private:
 # include <boost/asio/detail/impl/reactive_descriptor_service.ipp>
 #endif // defined(BOOST_ASIO_HEADER_ONLY)
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_REACTIVE_DESCRIPTOR_SERVICE_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/reactive_serial_port_service.hpp	2013-04-17 17:53:31.184205200 -0500
+++ src/boost_1_53_0/boost/asio/detail/reactive_serial_port_service.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -19,7 +19,7 @@
 #include <boost/asio/detail/config.hpp>
 
 #if defined(BOOST_ASIO_HAS_SERIAL_PORT)
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 #include <string>
 #include <boost/asio/error.hpp>
@@ -230,7 +230,7 @@ private:
 # include <boost/asio/detail/impl/reactive_serial_port_service.ipp>
 #endif // defined(BOOST_ASIO_HEADER_ONLY)
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 #endif // defined(BOOST_ASIO_HAS_SERIAL_PORT)
 
 #endif // BOOST_ASIO_DETAIL_REACTIVE_SERIAL_PORT_SERVICE_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/select_interrupter.hpp	2013-04-17 17:53:31.192205700 -0500
+++ src/boost_1_53_0/boost/asio/detail/select_interrupter.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__) || defined(__SYMBIAN32__)
+#if defined(BOOST_WINDOWS) || defined(__SYMBIAN32__)
 # include <boost/asio/detail/socket_select_interrupter.hpp>
 #elif defined(BOOST_ASIO_HAS_EVENTFD)
 # include <boost/asio/detail/eventfd_select_interrupter.hpp>
@@ -29,7 +29,7 @@ namespace boost {
 namespace asio {
 namespace detail {
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__) || defined(__SYMBIAN32__)
+#if defined(BOOST_WINDOWS) || defined(__SYMBIAN32__)
 typedef socket_select_interrupter select_interrupter;
 #elif defined(BOOST_ASIO_HAS_EVENTFD)
 typedef eventfd_select_interrupter select_interrupter;
--- origsrc/boost_1_53_0/boost/asio/detail/select_reactor.hpp	2013-04-17 17:53:31.210206700 -0500
+++ src/boost_1_53_0/boost/asio/detail/select_reactor.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -52,13 +52,13 @@ class select_reactor
   : public boost::asio::detail::service_base<select_reactor>
 {
 public:
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   enum op_types { read_op = 0, write_op = 1, except_op = 2,
     max_select_ops = 3, connect_op = 3, max_ops = 4 };
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
   enum op_types { read_op = 0, write_op = 1, except_op = 2,
     max_select_ops = 3, connect_op = 1, max_ops = 3 };
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
   // Per-descriptor data.
   struct per_descriptor_data
--- origsrc/boost_1_53_0/boost/asio/detail/signal_blocker.hpp	2013-04-17 17:53:31.227207700 -0500
+++ src/boost_1_53_0/boost/asio/detail/signal_blocker.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -18,7 +18,7 @@
 #include <boost/asio/detail/config.hpp>
 
 #if !defined(BOOST_HAS_THREADS) || defined(BOOST_ASIO_DISABLE_THREADS) \
-  || defined(BOOST_WINDOWS) || defined(__CYGWIN__) || defined(__SYMBIAN32__)
+  || defined(BOOST_WINDOWS) || defined(__SYMBIAN32__)
 # include <boost/asio/detail/null_signal_blocker.hpp>
 #elif defined(BOOST_HAS_PTHREADS)
 # include <boost/asio/detail/posix_signal_blocker.hpp>
@@ -31,7 +31,7 @@ namespace asio {
 namespace detail {
 
 #if !defined(BOOST_HAS_THREADS) || defined(BOOST_ASIO_DISABLE_THREADS) \
-  || defined(BOOST_WINDOWS) || defined(__CYGWIN__) || defined(__SYMBIAN32__)
+  || defined(BOOST_WINDOWS) || defined(__SYMBIAN32__)
 typedef null_signal_blocker signal_blocker;
 #elif defined(BOOST_HAS_PTHREADS)
 typedef posix_signal_blocker signal_blocker;
--- origsrc/boost_1_53_0/boost/asio/detail/signal_init.hpp	2013-04-17 17:53:31.244208700 -0500
+++ src/boost_1_53_0/boost/asio/detail/signal_init.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 #include <csignal>
 
@@ -44,6 +44,6 @@ public:
 
 #include <boost/asio/detail/pop_options.hpp>
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_SIGNAL_INIT_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/signal_set_service.hpp	2013-04-17 17:53:31.262209700 -0500
+++ src/boost_1_53_0/boost/asio/detail/signal_set_service.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -27,9 +27,9 @@
 #include <boost/asio/detail/signal_op.hpp>
 #include <boost/asio/detail/socket_types.hpp>
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 # include <boost/asio/detail/reactor.hpp>
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 #include <boost/asio/detail/push_options.hpp>
 
@@ -181,7 +181,7 @@ private:
   // The io_service instance used for dispatching handlers.
   io_service_impl& io_service_;
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
   // The type used for registering for pipe reactor notifications.
   class pipe_read_op;
 
@@ -190,7 +190,7 @@ private:
 
   // The per-descriptor reactor data used for the pipe.
   reactor::per_descriptor_data reactor_data_;
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
   // A mapping from signal number to the registered signal sets.
   registration* registrations_[max_signal_number];
--- origsrc/boost_1_53_0/boost/asio/detail/socket_ops.hpp	2013-04-17 17:53:31.279210700 -0500
+++ src/boost_1_53_0/boost/asio/detail/socket_ops.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -119,11 +119,11 @@ BOOST_ASIO_DECL size_t available(socket_
 BOOST_ASIO_DECL int listen(socket_type s,
     int backlog, boost::system::error_code& ec);
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 typedef WSABUF buf;
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
 typedef iovec buf;
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 BOOST_ASIO_DECL void init_buf(buf& b, void* data, size_t size);
 
--- origsrc/boost_1_53_0/boost/asio/detail/socket_select_interrupter.hpp	2013-04-17 17:53:31.296211600 -0500
+++ src/boost_1_53_0/boost/asio/detail/socket_select_interrupter.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -18,7 +18,7 @@
 #include <boost/asio/detail/config.hpp>
 
 #if defined(BOOST_WINDOWS) \
-  || defined(__CYGWIN__) \
+  \
   || defined(__SYMBIAN32__)
 
 #include <boost/asio/detail/socket_types.hpp>
@@ -83,7 +83,7 @@ private:
 #endif // defined(BOOST_ASIO_HEADER_ONLY)
 
 #endif // defined(BOOST_WINDOWS)
-       // || defined(__CYGWIN__)
+       //
        // || defined(__SYMBIAN32__)
 
 #endif // BOOST_ASIO_DETAIL_SOCKET_SELECT_INTERRUPTER_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/socket_types.hpp	2013-04-17 17:53:31.312212600 -0500
+++ src/boost_1_53_0/boost/asio/detail/socket_types.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # if defined(_WINSOCKAPI_) && !defined(_WINSOCK2API_)
 #  error WinSock.h has already been included
 # endif // defined(_WINSOCKAPI_) && !defined(_WINSOCK2API_)
@@ -81,7 +81,7 @@ namespace boost {
 namespace asio {
 namespace detail {
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 typedef SOCKET socket_type;
 const SOCKET invalid_socket = INVALID_SOCKET;
 const int socket_error_retval = SOCKET_ERROR;
--- origsrc/boost_1_53_0/boost/asio/detail/win_fd_set_adapter.hpp	2013-04-17 17:53:31.317212800 -0500
+++ src/boost_1_53_0/boost/asio/detail/win_fd_set_adapter.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 
 #include <boost/asio/detail/noncopyable.hpp>
 #include <boost/asio/detail/socket_types.hpp>
@@ -120,6 +120,6 @@ private:
 
 #include <boost/asio/detail/pop_options.hpp>
 
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_WIN_FD_SET_ADAPTER_HPP
--- origsrc/boost_1_53_0/boost/asio/detail/winsock_init.hpp	2013-04-17 17:53:31.322213100 -0500
+++ src/boost_1_53_0/boost/asio/detail/winsock_init.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -17,7 +17,7 @@
 
 #include <boost/asio/detail/config.hpp>
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 
 #include <boost/asio/detail/push_options.hpp>
 
@@ -87,6 +87,6 @@ static const winsock_init<>& winsock_ini
 # include <boost/asio/detail/impl/winsock_init.ipp>
 #endif // defined(BOOST_ASIO_HEADER_ONLY)
 
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
 
 #endif // BOOST_ASIO_DETAIL_WINSOCK_INIT_HPP
--- origsrc/boost_1_53_0/boost/asio/error.hpp	2013-04-17 17:53:31.327213400 -0500
+++ src/boost_1_53_0/boost/asio/error.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -18,7 +18,7 @@
 #include <boost/asio/detail/config.hpp>
 #include <boost/cerrno.hpp>
 #include <boost/system/error_code.hpp>
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # include <winerror.h>
 #else
 # include <cerrno>
@@ -36,7 +36,7 @@
 # define BOOST_ASIO_GETADDRINFO_ERROR(e) implementation_defined
 /// INTERNAL ONLY.
 # define BOOST_ASIO_WIN_OR_POSIX(e_win, e_posix) implementation_defined
-#elif defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#elif defined(BOOST_WINDOWS)
 # define BOOST_ASIO_NATIVE_ERROR(e) e
 # define BOOST_ASIO_SOCKET_ERROR(e) WSA ## e
 # define BOOST_ASIO_NETDB_ERROR(e) WSA ## e
@@ -216,7 +216,7 @@ inline const boost::system::error_catego
   return boost::system::system_category();
 }
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 extern BOOST_ASIO_DECL
 const boost::system::error_category& get_netdb_category();
@@ -224,7 +224,7 @@ const boost::system::error_category& get
 extern BOOST_ASIO_DECL
 const boost::system::error_category& get_addrinfo_category();
 
-#else // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#else // !defined(BOOST_WINDOWS)
 
 inline const boost::system::error_category& get_netdb_category()
 {
@@ -236,7 +236,7 @@ inline const boost::system::error_catego
   return get_system_category();
 }
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 extern BOOST_ASIO_DECL
 const boost::system::error_category& get_misc_category();
--- origsrc/boost_1_53_0/boost/asio/impl/error.ipp	2013-04-17 17:53:31.332213700 -0500
+++ src/boost_1_53_0/boost/asio/impl/error.ipp	2013-04-17 17:21:03.000000000 -0500
@@ -24,7 +24,7 @@ namespace boost {
 namespace asio {
 namespace error {
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 
 namespace detail {
 
@@ -86,7 +86,7 @@ const boost::system::error_category& get
   return instance;
 }
 
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 namespace detail {
 
--- origsrc/boost_1_53_0/boost/asio/impl/serial_port_base.ipp	2013-04-17 17:53:31.338214000 -0500
+++ src/boost_1_53_0/boost/asio/impl/serial_port_base.ipp	2013-04-17 17:21:03.000000000 -0500
@@ -27,7 +27,7 @@
 
 #if defined(GENERATING_DOCUMENTATION)
 # define BOOST_ASIO_OPTION_STORAGE implementation_defined
-#elif defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#elif defined(BOOST_WINDOWS)
 # define BOOST_ASIO_OPTION_STORAGE DCB
 #else
 # define BOOST_ASIO_OPTION_STORAGE termios
@@ -41,7 +41,7 @@ namespace asio {
 boost::system::error_code serial_port_base::baud_rate::store(
     BOOST_ASIO_OPTION_STORAGE& storage, boost::system::error_code& ec) const
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   storage.BaudRate = value_;
 #else
   speed_t baud;
@@ -129,7 +129,7 @@ boost::system::error_code serial_port_ba
 boost::system::error_code serial_port_base::baud_rate::load(
     const BOOST_ASIO_OPTION_STORAGE& storage, boost::system::error_code& ec)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   value_ = storage.BaudRate;
 #else
   speed_t baud = ::cfgetospeed(&storage);
@@ -222,7 +222,7 @@ serial_port_base::flow_control::flow_con
 boost::system::error_code serial_port_base::flow_control::store(
     BOOST_ASIO_OPTION_STORAGE& storage, boost::system::error_code& ec) const
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   storage.fOutxCtsFlow = FALSE;
   storage.fOutxDsrFlow = FALSE;
   storage.fTXContinueOnXoff = TRUE;
@@ -289,7 +289,7 @@ boost::system::error_code serial_port_ba
 boost::system::error_code serial_port_base::flow_control::load(
     const BOOST_ASIO_OPTION_STORAGE& storage, boost::system::error_code& ec)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   if (storage.fOutX && storage.fInX)
   {
     value_ = software;
@@ -340,7 +340,7 @@ serial_port_base::parity::parity(serial_
 boost::system::error_code serial_port_base::parity::store(
     BOOST_ASIO_OPTION_STORAGE& storage, boost::system::error_code& ec) const
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   switch (value_)
   {
   case none:
@@ -387,7 +387,7 @@ boost::system::error_code serial_port_ba
 boost::system::error_code serial_port_base::parity::load(
     const BOOST_ASIO_OPTION_STORAGE& storage, boost::system::error_code& ec)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   if (storage.Parity == EVENPARITY)
   {
     value_ = even;
@@ -435,7 +435,7 @@ serial_port_base::stop_bits::stop_bits(
 boost::system::error_code serial_port_base::stop_bits::store(
     BOOST_ASIO_OPTION_STORAGE& storage, boost::system::error_code& ec) const
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   switch (value_)
   {
   case one:
@@ -471,7 +471,7 @@ boost::system::error_code serial_port_ba
 boost::system::error_code serial_port_base::stop_bits::load(
     const BOOST_ASIO_OPTION_STORAGE& storage, boost::system::error_code& ec)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   if (storage.StopBits == ONESTOPBIT)
   {
     value_ = one;
@@ -508,7 +508,7 @@ serial_port_base::character_size::charac
 boost::system::error_code serial_port_base::character_size::store(
     BOOST_ASIO_OPTION_STORAGE& storage, boost::system::error_code& ec) const
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   storage.ByteSize = value_;
 #else
   storage.c_cflag &= ~CSIZE;
@@ -528,7 +528,7 @@ boost::system::error_code serial_port_ba
 boost::system::error_code serial_port_base::character_size::load(
     const BOOST_ASIO_OPTION_STORAGE& storage, boost::system::error_code& ec)
 {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   value_ = storage.ByteSize;
 #else
   if ((storage.c_cflag & CSIZE) == CS5) { value_ = 5; }
--- origsrc/boost_1_53_0/boost/asio/io_service.hpp	2013-04-17 17:53:31.344214400 -0500
+++ src/boost_1_53_0/boost/asio/io_service.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -30,7 +30,7 @@
 # include <boost/asio/detail/task_io_service_fwd.hpp>
 #endif
 
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
 # include <boost/asio/detail/winsock_init.hpp>
 #elif defined(__sun) || defined(__QNX__) || defined(__hpux) || defined(_AIX) \
   || defined(__osf__)
@@ -600,7 +600,7 @@ public:
   friend bool has_service(io_service& ios);
 
 private:
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
   detail::winsock_init<> init_;
 #elif defined(__sun) || defined(__QNX__) || defined(__hpux) || defined(_AIX) \
   || defined(__osf__)
--- origsrc/boost_1_53_0/boost/asio/serial_port_base.hpp	2013-04-17 17:53:31.349214700 -0500
+++ src/boost_1_53_0/boost/asio/serial_port_base.hpp	2013-04-17 17:21:03.000000000 -0500
@@ -21,9 +21,9 @@
 #if defined(BOOST_ASIO_HAS_SERIAL_PORT) \
   || defined(GENERATING_DOCUMENTATION)
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
 # include <termios.h>
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 
 #include <boost/detail/workaround.hpp>
 #include <boost/asio/detail/socket_types.hpp>
@@ -31,7 +31,7 @@
 
 #if defined(GENERATING_DOCUMENTATION)
 # define BOOST_ASIO_OPTION_STORAGE implementation_defined
-#elif defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#elif defined(BOOST_WINDOWS)
 # define BOOST_ASIO_OPTION_STORAGE DCB
 #else
 # define BOOST_ASIO_OPTION_STORAGE termios
--- origsrc/boost_1_53_0/boost/asio/ssl/detail/impl/openssl_init.ipp	2013-04-17 17:53:31.356215100 -0500
+++ src/boost_1_53_0/boost/asio/ssl/detail/impl/openssl_init.ipp	2013-04-17 17:21:03.000000000 -0500
@@ -64,15 +64,15 @@ public:
 private:
   static unsigned long openssl_id_func()
   {
-#if defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#if defined(BOOST_WINDOWS)
     return ::GetCurrentThreadId();
-#else // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#else // defined(BOOST_WINDOWS)
     void* id = instance()->thread_id_;
     if (id == 0)
       instance()->thread_id_ = id = &id; // Ugh.
     BOOST_ASSERT(sizeof(unsigned long) >= sizeof(void*));
     return reinterpret_cast<unsigned long>(id);
-#endif // defined(BOOST_WINDOWS) || defined(__CYGWIN__)
+#endif // defined(BOOST_WINDOWS)
   }
 
   static void openssl_locking_func(int mode, int n, 
@@ -88,10 +88,10 @@ private:
   std::vector<boost::asio::detail::shared_ptr<
         boost::asio::detail::mutex> > mutexes_;
 
-#if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#if !defined(BOOST_WINDOWS)
   // The thread identifiers to be used by openssl.
   boost::asio::detail::tss_ptr<void> thread_id_;
-#endif // !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
+#endif // !defined(BOOST_WINDOWS)
 };
 
 boost::asio::detail::shared_ptr<openssl_init_base::do_init>
